// @radixdlt/babylon-gateway-api-sdk/dist/babylon-gateway-api-sdk.mjs
var Ae = Object.defineProperty;
var qe = (e, t, n) => t in e ? Ae(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n;
var m = (e, t, n) => (qe(e, typeof t != "symbol" ? t + "" : t, n), n);
const Pe = "https://rcnet.radixdlt.com".replace(/\/+$/, "");
class Ce {
  constructor(t = {}) {
    this.configuration = t;
  }
  set config(t) {
    this.configuration = t;
  }
  get basePath() {
    return this.configuration.basePath != null ? this.configuration.basePath : Pe;
  }
  get fetchApi() {
    return this.configuration.fetchApi;
  }
  get middleware() {
    return this.configuration.middleware || [];
  }
  get queryParamsStringify() {
    return this.configuration.queryParamsStringify || M;
  }
  get username() {
    return this.configuration.username;
  }
  get password() {
    return this.configuration.password;
  }
  get apiKey() {
    const t = this.configuration.apiKey;
    if (t)
      return typeof t == "function" ? t : () => t;
  }
  get accessToken() {
    const t = this.configuration.accessToken;
    if (t)
      return typeof t == "function" ? t : async () => t;
  }
  get headers() {
    return this.configuration.headers;
  }
  get credentials() {
    return this.configuration.credentials;
  }
}
const xe = new Ce();
class O {
  constructor(t = xe) {
    m(this, "middleware");
    m(this, "fetchApi", async (t, n) => {
      let i = { url: t, init: n };
      for (const s of this.middleware)
        s.pre && (i = await s.pre({
          fetch: this.fetchApi,
          ...i
        }) || i);
      let o;
      try {
        o = await (this.configuration.fetchApi || fetch)(i.url, i.init);
      } catch (s) {
        for (const a of this.middleware)
          a.onError && (o = await a.onError({
            fetch: this.fetchApi,
            url: i.url,
            init: i.init,
            error: s,
            response: o ? o.clone() : void 0
          }) || o);
        if (o === void 0)
          throw s;
      }
      for (const s of this.middleware)
        s.post && (o = await s.post({
          fetch: this.fetchApi,
          url: i.url,
          init: i.init,
          response: o.clone()
        }) || o);
      return o;
    });
    this.configuration = t, this.middleware = t.middleware;
  }
  withMiddleware(...t) {
    const n = this.clone();
    return n.middleware = n.middleware.concat(...t), n;
  }
  withPreMiddleware(...t) {
    const n = t.map((i) => ({ pre: i }));
    return this.withMiddleware(...n);
  }
  withPostMiddleware(...t) {
    const n = t.map((i) => ({ post: i }));
    return this.withMiddleware(...n);
  }
  async request(t, n) {
    const { url: i, init: o } = await this.createFetchParams(t, n), s = await this.fetchApi(i, o);
    if (s && s.status >= 200 && s.status < 300)
      return s;
    throw await S.from(s);
  }
  async createFetchParams(t, n) {
    let i = this.configuration.basePath + t.path;
    t.query !== void 0 && Object.keys(t.query).length !== 0 && (i += "?" + this.configuration.queryParamsStringify(t.query));
    const o = Object.assign({}, this.configuration.headers, t.headers);
    Object.keys(o).forEach((L) => o[L] === void 0 ? delete o[L] : {});
    const s = typeof n == "function" ? n : async () => n, a = {
      method: t.method,
      headers: o,
      body: t.body,
      credentials: this.configuration.credentials
    }, p = {
      ...a,
      ...await s({
        init: a,
        context: t
      })
    }, De = {
      ...p,
      body: ke(p.body) || p.body instanceof URLSearchParams || Ve(p.body) ? p.body : JSON.stringify(p.body)
    };
    return { url: i, init: De };
  }
  clone() {
    const t = this.constructor, n = new t(this.configuration);
    return n.middleware = this.middleware.slice(), n;
  }
}
function Ve(e) {
  return typeof Blob < "u" && e instanceof Blob;
}
function ke(e) {
  return typeof FormData < "u" && e instanceof FormData;
}
class S extends Error {
  constructor(n, i, o) {
    super(o ? JSON.stringify(o) : "Unknown error occurred");
    m(this, "name", "ResponseError");
    this.fetchResponse = n, this.status = i, this.errorResponse = o;
  }
  static async from(n) {
    const i = n.status;
    try {
      const o = await n.json();
      return new S(n, i, o);
    } catch {
      return new S(n, i, void 0);
    }
  }
}
class d extends Error {
  constructor(n, i) {
    super(i);
    m(this, "name", "RequiredError");
    this.field = n;
  }
}
const wi = {
  csv: ",",
  ssv: " ",
  tsv: "	",
  pipes: "|"
};
function r(e, t) {
  const n = e[t];
  return n != null;
}
function M(e, t = "") {
  return Object.keys(e).map((n) => G(n, e[n], t)).filter((n) => n.length > 0).join("&");
}
function G(e, t, n = "") {
  const i = n + (n.length ? `[${e}]` : e);
  if (t instanceof Array) {
    const o = t.map((s) => encodeURIComponent(String(s))).join(`&${encodeURIComponent(i)}=`);
    return `${encodeURIComponent(i)}=${o}`;
  }
  if (t instanceof Set) {
    const o = Array.from(t);
    return G(e, o, n);
  }
  return t instanceof Date ? `${encodeURIComponent(i)}=${encodeURIComponent(t.toISOString())}` : t instanceof Object ? M(t, i) : `${encodeURIComponent(i)}=${encodeURIComponent(String(t))}`;
}
function vi(e, t) {
  return Object.keys(e).reduce(
    (n, i) => ({ ...n, [i]: t(e[i]) }),
    {}
  );
}
function Di(e) {
  for (const t of e)
    if (t.contentType === "multipart/form-data")
      return !0;
  return !1;
}
class u {
  constructor(t, n = (i) => i) {
    this.raw = t, this.transformer = n;
  }
  async value() {
    return this.transformer(await this.raw.json());
  }
}
class Ai {
  constructor(t) {
    this.raw = t;
  }
  async value() {
  }
}
class qi {
  constructor(t) {
    this.raw = t;
  }
  async value() {
    return await this.raw.blob();
  }
}
class Pi {
  constructor(t) {
    this.raw = t;
  }
  async value() {
    return await this.raw.text();
  }
}
function Ci(e) {
  return !0;
}
function _(e) {
  return Le(e);
}
function Le(e, t) {
  return e == null ? e : {
    state_version: r(e, "state_version") ? e.state_version : void 0,
    timestamp: r(e, "timestamp") ? e.timestamp === null ? null : new Date(e.timestamp) : void 0,
    epoch: r(e, "epoch") ? e.epoch : void 0,
    round: r(e, "round") ? e.round : void 0
  };
}
function g(e) {
  if (e !== void 0)
    return e === null ? null : {
      state_version: e.state_version,
      timestamp: e.timestamp === void 0 ? void 0 : e.timestamp === null ? null : e.timestamp.toISOString(),
      epoch: e.epoch,
      round: e.round
    };
}
function xi(e) {
  return !0;
}
function Vi(e) {
  return Me(e);
}
function Me(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0
  };
}
function ki(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state)
    };
}
function Li(e) {
  let t = !0;
  return t = t && "value" in e, t;
}
function Ge(e) {
  return Ke(e);
}
function Ke(e, t) {
  return e == null ? e : {
    value: e.value,
    address: r(e, "address") ? e.address : void 0
  };
}
function Ue(e) {
  if (e !== void 0)
    return e === null ? null : {
      value: e.value,
      address: e.address
    };
}
const Mi = {
  Unknown: "Unknown",
  CommittedSuccess: "CommittedSuccess",
  CommittedFailure: "CommittedFailure",
  Pending: "Pending",
  Rejected: "Rejected"
};
function N(e) {
  return Be(e);
}
function Be(e, t) {
  return e;
}
function Gi(e) {
  return e;
}
function Ki(e) {
  let t = !0;
  return t = t && "state_version" in e, t = t && "transaction_status" in e, t;
}
function E(e) {
  return $e(e);
}
function $e(e, t) {
  return e == null ? e : {
    state_version: e.state_version,
    transaction_status: N(e.transaction_status),
    payload_hash_hex: r(e, "payload_hash_hex") ? e.payload_hash_hex : void 0,
    intent_hash_hex: r(e, "intent_hash_hex") ? e.intent_hash_hex : void 0,
    fee_paid: r(e, "fee_paid") ? Ge(e.fee_paid) : void 0,
    confirmed_at: r(e, "confirmed_at") ? e.confirmed_at === null ? null : new Date(e.confirmed_at) : void 0,
    error_message: r(e, "error_message") ? e.error_message : void 0
  };
}
function T(e) {
  if (e !== void 0)
    return e === null ? null : {
      state_version: e.state_version,
      transaction_status: e.transaction_status,
      payload_hash_hex: e.payload_hash_hex,
      intent_hash_hex: e.intent_hash_hex,
      fee_paid: Ue(e.fee_paid),
      confirmed_at: e.confirmed_at === void 0 ? void 0 : e.confirmed_at === null ? null : e.confirmed_at.toISOString(),
      error_message: e.error_message
    };
}
function Ui(e) {
  return !0;
}
function Bi(e) {
  return ze(e);
}
function ze(e, t) {
  return e == null ? e : {
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0
  };
}
function $i(e) {
  if (e !== void 0)
    return e === null ? null : {
      cursor: e.cursor,
      limit_per_page: e.limit_per_page
    };
}
function zi(e) {
  let t = !0;
  return t = t && "raw_hex" in e, t = t && "raw_json" in e, t;
}
function We(e) {
  return He(e);
}
function He(e, t) {
  return e == null ? e : {
    raw_hex: e.raw_hex,
    raw_json: e.raw_json,
    as_string: r(e, "as_string") ? e.as_string : void 0,
    as_string_collection: r(e, "as_string_collection") ? e.as_string_collection : void 0
  };
}
function Qe(e) {
  if (e !== void 0)
    return e === null ? null : {
      raw_hex: e.raw_hex,
      raw_json: e.raw_json,
      as_string: e.as_string,
      as_string_collection: e.as_string_collection
    };
}
function Wi(e) {
  let t = !0;
  return t = t && "key" in e, t = t && "value" in e, t = t && "last_updated_at_state_version" in e, t;
}
function I(e) {
  return Xe(e);
}
function Xe(e, t) {
  return e == null ? e : {
    key: e.key,
    value: We(e.value),
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function J(e) {
  if (e !== void 0)
    return e === null ? null : {
      key: e.key,
      value: Qe(e.value),
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function Hi(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function K(e) {
  return Ye(e);
}
function Ye(e, t) {
  return e == null ? e : {
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(I)
  };
}
function U(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(J)
    };
}
function Qi(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Xi(e) {
  return Ze(e);
}
function Ze(e, t) {
  return e == null ? e : {
    items: e.items.map(I)
  };
}
function Yi(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(J)
    };
}
function Zi(e) {
  return !0;
}
function ji(e) {
  return je(e);
}
function je(e, t) {
  return e == null ? e : {
    as_string: r(e, "as_string") ? e.as_string : void 0,
    as_string_collection: r(e, "as_string_collection") ? e.as_string_collection : void 0
  };
}
function eo(e) {
  if (e !== void 0)
    return e === null ? null : {
      as_string: e.as_string,
      as_string_collection: e.as_string_collection
    };
}
const to = {
  EntityNotFoundError: "EntityNotFoundError"
};
function ro(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "address" in e, t;
}
function no(e) {
  return B(e);
}
function B(e, t) {
  return e == null ? e : {
    type: e.type,
    address: e.address
  };
}
function et(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      address: e.address
    };
}
const io = {
  EntityNotFoundError: "EntityNotFoundError"
};
function oo(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function so(e) {
  return tt(e);
}
function tt(e, t) {
  return e == null ? e : {
    address: e.address,
    type: r(e, "type") ? e.type : void 0
  };
}
function ao(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      type: e.type
    };
}
const uo = {
  InternalServerError: "InternalServerError"
};
function lo(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "exception" in e, t = t && "cause" in e, t;
}
function co(e) {
  return $(e);
}
function $(e, t) {
  return e == null ? e : {
    type: e.type,
    exception: e.exception,
    cause: e.cause
  };
}
function rt(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      exception: e.exception,
      cause: e.cause
    };
}
const _o = {
  InvalidEntityError: "InvalidEntityError"
};
function go(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "address" in e, t;
}
function fo(e) {
  return z(e);
}
function z(e, t) {
  return e == null ? e : {
    type: e.type,
    address: e.address
  };
}
function nt(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      address: e.address
    };
}
function po(e) {
  let t = !0;
  return t = t && "path" in e, t = t && "errors" in e, t;
}
function W(e) {
  return it(e);
}
function it(e, t) {
  return e == null ? e : {
    path: e.path,
    errors: e.errors
  };
}
function H(e) {
  if (e !== void 0)
    return e === null ? null : {
      path: e.path,
      errors: e.errors
    };
}
const mo = {
  InvalidRequestError: "InvalidRequestError"
};
function yo(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "validation_errors" in e, t;
}
function So(e) {
  return Q(e);
}
function Q(e, t) {
  return e == null ? e : {
    type: e.type,
    validation_errors: e.validation_errors.map(W)
  };
}
function ot(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      validation_errors: e.validation_errors.map(H)
    };
}
const Oo = {
  InvalidTransactionError: "InvalidTransactionError"
};
function No(e) {
  let t = !0;
  return t = t && "type" in e, t;
}
function Fo(e) {
  return X(e);
}
function X(e, t) {
  return e == null ? e : {
    type: e.type
  };
}
function st(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
const Ro = {
  NotSyncedUpError: "NotSyncedUpError"
};
function bo(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "request_type" in e, t = t && "current_sync_delay_seconds" in e, t = t && "max_allowed_sync_delay_seconds" in e, t;
}
function ho(e) {
  return Y(e);
}
function Y(e, t) {
  return e == null ? e : {
    type: e.type,
    request_type: e.request_type,
    current_sync_delay_seconds: e.current_sync_delay_seconds,
    max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
  };
}
function at(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      request_type: e.request_type,
      current_sync_delay_seconds: e.current_sync_delay_seconds,
      max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
    };
}
const Eo = {
  TransactionNotFoundError: "TransactionNotFoundError"
};
function To(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "intent_hash_hex" in e, t;
}
function Io(e) {
  return Z(e);
}
function Z(e, t) {
  return e == null ? e : {
    type: e.type,
    intent_hash_hex: e.intent_hash_hex
  };
}
function ut(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      intent_hash_hex: e.intent_hash_hex
    };
}
function lt(e) {
  return ct(e);
}
function ct(e, t) {
  if (e == null)
    return e;
  switch (e.type) {
    case "EntityNotFoundError":
      return { ...B(e), type: "EntityNotFoundError" };
    case "InternalServerError":
      return { ...$(e), type: "InternalServerError" };
    case "InvalidEntityError":
      return { ...z(e), type: "InvalidEntityError" };
    case "InvalidRequestError":
      return { ...Q(e), type: "InvalidRequestError" };
    case "InvalidTransactionError":
      return { ...X(e), type: "InvalidTransactionError" };
    case "NotSyncedUpError":
      return { ...Y(e), type: "NotSyncedUpError" };
    case "TransactionNotFoundError":
      return { ...Z(e), type: "TransactionNotFoundError" };
    default:
      throw new Error(`No variant of GatewayError exists with 'type=${e.type}'`);
  }
}
function dt(e) {
  if (e !== void 0) {
    if (e === null)
      return null;
    switch (e.type) {
      case "EntityNotFoundError":
        return et(e);
      case "InternalServerError":
        return rt(e);
      case "InvalidEntityError":
        return nt(e);
      case "InvalidRequestError":
        return ot(e);
      case "InvalidTransactionError":
        return st(e);
      case "NotSyncedUpError":
        return at(e);
      case "TransactionNotFoundError":
        return ut(e);
      default:
        throw new Error(`No variant of GatewayError exists with 'type=${e.type}'`);
    }
  }
}
function Jo(e) {
  let t = !0;
  return t = t && "message" in e, t;
}
function wo(e) {
  return _t(e);
}
function _t(e, t) {
  return e == null ? e : {
    message: e.message,
    code: r(e, "code") ? e.code : void 0,
    details: r(e, "details") ? lt(e.details) : void 0,
    trace_id: r(e, "trace_id") ? e.trace_id : void 0
  };
}
function vo(e) {
  if (e !== void 0)
    return e === null ? null : {
      message: e.message,
      code: e.code,
      details: dt(e.details),
      trace_id: e.trace_id
    };
}
const Do = {
  Global: "Global"
};
function Ao(e) {
  let t = !0;
  return t = t && "aggregation_level" in e, t = t && "resource_address" in e, t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function F(e) {
  return j(e);
}
function j(e, t) {
  return e == null ? e : {
    aggregation_level: e.aggregation_level,
    resource_address: e.resource_address,
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function y(e) {
  if (e !== void 0)
    return e === null ? null : {
      aggregation_level: e.aggregation_level,
      resource_address: e.resource_address,
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function qo(e) {
  let t = !0;
  return t = t && "vault_address" in e, t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function w(e) {
  return gt(e);
}
function gt(e, t) {
  return e == null ? e : {
    vault_address: e.vault_address,
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function v(e) {
  if (e !== void 0)
    return e === null ? null : {
      vault_address: e.vault_address,
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function Po(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function ee(e) {
  return ft(e);
}
function ft(e, t) {
  return e == null ? e : {
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(w)
  };
}
function te(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(v)
    };
}
const Co = {
  Vault: "Vault"
};
function xo(e) {
  let t = !0;
  return t = t && "aggregation_level" in e, t = t && "resource_address" in e, t = t && "vaults" in e, t;
}
function Vo(e) {
  return re(e);
}
function re(e, t) {
  return e == null ? e : {
    aggregation_level: e.aggregation_level,
    resource_address: e.resource_address,
    vaults: ee(e.vaults)
  };
}
function pt(e) {
  if (e !== void 0)
    return e === null ? null : {
      aggregation_level: e.aggregation_level,
      resource_address: e.resource_address,
      vaults: te(e.vaults)
    };
}
function D(e) {
  return mt(e);
}
function mt(e, t) {
  if (e == null)
    return e;
  switch (e.aggregation_level) {
    case "Global":
      return { ...j(e), aggregation_level: "Global" };
    case "Vault":
      return { ...re(e), aggregation_level: "Vault" };
    default:
      throw new Error(`No variant of FungibleResourcesCollectionItem exists with 'aggregation_level=${e.aggregation_level}'`);
  }
}
function A(e) {
  if (e !== void 0) {
    if (e === null)
      return null;
    switch (e.aggregation_level) {
      case "Global":
        return y(e);
      case "Vault":
        return pt(e);
      default:
        throw new Error(`No variant of FungibleResourcesCollectionItem exists with 'aggregation_level=${e.aggregation_level}'`);
    }
  }
}
function ko(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function yt(e) {
  return St(e);
}
function St(e, t) {
  return e == null ? e : {
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(D)
  };
}
function Ot(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(A)
    };
}
function Lo(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Mo(e) {
  return Nt(e);
}
function Nt(e, t) {
  return e == null ? e : {
    items: e.items.map(D)
  };
}
function Go(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(A)
    };
}
const Ko = {
  Global: "Global",
  Vault: "Vault"
};
function f(e) {
  return Ft(e);
}
function Ft(e, t) {
  return e;
}
function Uo(e) {
  return e;
}
function Bo(e) {
  let t = !0;
  return t = t && "aggregation_level" in e, t = t && "resource_address" in e, t;
}
function $o(e) {
  return Rt(e);
}
function Rt(e, t) {
  return e == null ? e : {
    aggregation_level: f(e.aggregation_level),
    resource_address: e.resource_address
  };
}
function zo(e) {
  if (e !== void 0)
    return e === null ? null : {
      aggregation_level: e.aggregation_level,
      resource_address: e.resource_address
    };
}
const Wo = {
  Global: "Global"
};
function Ho(e) {
  let t = !0;
  return t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function Qo(e) {
  return bt(e);
}
function bt(e, t) {
  return e == null ? e : {
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version,
    aggregation_level: r(e, "aggregation_level") ? e.aggregation_level : void 0
  };
}
function Xo(e) {
  if (e !== void 0)
    return e === null ? null : {
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version,
      aggregation_level: e.aggregation_level
    };
}
const Yo = {
  Vault: "Vault"
};
function Zo(e) {
  let t = !0;
  return t = t && "vaults" in e, t;
}
function jo(e) {
  return ht(e);
}
function ht(e, t) {
  return e == null ? e : {
    vaults: ee(e.vaults),
    aggregation_level: r(e, "aggregation_level") ? e.aggregation_level : void 0
  };
}
function es(e) {
  if (e !== void 0)
    return e === null ? null : {
      vaults: te(e.vaults),
      aggregation_level: e.aggregation_level
    };
}
function ts(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function rs(e) {
  return Et(e);
}
function Et(e, t) {
  return e == null ? e : {
    items: e.items.map(w)
  };
}
function ns(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(v)
    };
}
function is(e) {
  let t = !0;
  return t = t && "type" in e, t;
}
function os(e) {
  return Tt(e);
}
function Tt(e, t) {
  return e == null ? e : {
    type: e.type
  };
}
function ss(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
function as(e) {
  let t = !0;
  return t = t && "state_version" in e, t;
}
function us(e) {
  return It(e);
}
function It(e, t) {
  return e == null ? e : {
    state_version: e.state_version
  };
}
function ls(e) {
  if (e !== void 0)
    return e === null ? null : {
      state_version: e.state_version
    };
}
function cs(e) {
  let t = !0;
  return t = t && "release_version" in e, t = t && "open_api_schema_version" in e, t;
}
function ne(e) {
  return Jt(e);
}
function Jt(e, t) {
  return e == null ? e : {
    release_version: e.release_version,
    open_api_schema_version: e.open_api_schema_version
  };
}
function ie(e) {
  if (e !== void 0)
    return e === null ? null : {
      release_version: e.release_version,
      open_api_schema_version: e.open_api_schema_version
    };
}
function ds(e) {
  let t = !0;
  return t = t && "network" in e, t = t && "state_version" in e, t = t && "proposer_round_timestamp" in e, t = t && "epoch" in e, t = t && "round" in e, t;
}
function l(e) {
  return wt(e);
}
function wt(e, t) {
  return e == null ? e : {
    network: e.network,
    state_version: e.state_version,
    proposer_round_timestamp: e.proposer_round_timestamp,
    epoch: e.epoch,
    round: e.round
  };
}
function c(e) {
  if (e !== void 0)
    return e === null ? null : {
      network: e.network,
      state_version: e.state_version,
      proposer_round_timestamp: e.proposer_round_timestamp,
      epoch: e.epoch,
      round: e.round
    };
}
function _s(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "release_info" in e, t;
}
function vt(e) {
  return Dt(e);
}
function Dt(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    release_info: ne(e.release_info)
  };
}
function gs(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      release_info: ie(e.release_info)
    };
}
function fs(e) {
  let t = !0;
  return t = t && "release_info" in e, t;
}
function ps(e) {
  return At(e);
}
function At(e, t) {
  return e == null ? e : {
    release_info: ne(e.release_info)
  };
}
function ms(e) {
  if (e !== void 0)
    return e === null ? null : {
      release_info: ie(e.release_info)
    };
}
const ys = {
  InternalServerError: "InternalServerError"
};
function Ss(e) {
  let t = !0;
  return t = t && "exception" in e, t = t && "cause" in e, t;
}
function Os(e) {
  return qt(e);
}
function qt(e, t) {
  return e == null ? e : {
    exception: e.exception,
    cause: e.cause,
    type: r(e, "type") ? e.type : void 0
  };
}
function Ns(e) {
  if (e !== void 0)
    return e === null ? null : {
      exception: e.exception,
      cause: e.cause,
      type: e.type
    };
}
const Fs = {
  InvalidEntityError: "InvalidEntityError"
};
function Rs(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function bs(e) {
  return Pt(e);
}
function Pt(e, t) {
  return e == null ? e : {
    address: e.address,
    type: r(e, "type") ? e.type : void 0
  };
}
function hs(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      type: e.type
    };
}
const Es = {
  InvalidRequestError: "InvalidRequestError"
};
function Ts(e) {
  let t = !0;
  return t = t && "validation_errors" in e, t;
}
function Is(e) {
  return Ct(e);
}
function Ct(e, t) {
  return e == null ? e : {
    validation_errors: e.validation_errors.map(W),
    type: r(e, "type") ? e.type : void 0
  };
}
function Js(e) {
  if (e !== void 0)
    return e === null ? null : {
      validation_errors: e.validation_errors.map(H),
      type: e.type
    };
}
const ws = {
  InvalidTransactionError: "InvalidTransactionError"
};
function vs(e) {
  return !0;
}
function Ds(e) {
  return xt(e);
}
function xt(e, t) {
  return e == null ? e : {
    type: r(e, "type") ? e.type : void 0
  };
}
function As(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
function qs(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t;
}
function Ps(e) {
  return Vt(e);
}
function Vt(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state)
  };
}
function Cs(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state)
    };
}
function xs(e) {
  let t = !0;
  return t = t && "faucet" in e, t = t && "epoch_manager" in e, t = t && "clock" in e, t = t && "ecdsa_secp256k1" in e, t = t && "eddsa_ed25519" in e, t = t && "xrd" in e, t;
}
function kt(e) {
  return Lt(e);
}
function Lt(e, t) {
  return e == null ? e : {
    faucet: e.faucet,
    epoch_manager: e.epoch_manager,
    clock: e.clock,
    ecdsa_secp256k1: e.ecdsa_secp256k1,
    eddsa_ed25519: e.eddsa_ed25519,
    xrd: e.xrd
  };
}
function Mt(e) {
  if (e !== void 0)
    return e === null ? null : {
      faucet: e.faucet,
      epoch_manager: e.epoch_manager,
      clock: e.clock,
      ecdsa_secp256k1: e.ecdsa_secp256k1,
      eddsa_ed25519: e.eddsa_ed25519,
      xrd: e.xrd
    };
}
function Vs(e) {
  let t = !0;
  return t = t && "network_id" in e, t = t && "network_name" in e, t = t && "well_known_addresses" in e, t;
}
function Gt(e) {
  return Kt(e);
}
function Kt(e, t) {
  return e == null ? e : {
    network_id: e.network_id,
    network_name: e.network_name,
    well_known_addresses: kt(e.well_known_addresses)
  };
}
function ks(e) {
  if (e !== void 0)
    return e === null ? null : {
      network_id: e.network_id,
      network_name: e.network_name,
      well_known_addresses: Mt(e.well_known_addresses)
    };
}
const Ls = {
  String: "String",
  Integer: "Integer",
  Bytes: "Bytes",
  Uuid: "Uuid"
};
function R(e) {
  return Ut(e);
}
function Ut(e, t) {
  return e;
}
function Ms(e) {
  return e;
}
function Gs(e) {
  let t = !0;
  return t = t && "non_fungible_id" in e, t;
}
function q(e) {
  return Bt(e);
}
function Bt(e, t) {
  return e == null ? e : {
    non_fungible_id: e.non_fungible_id
  };
}
function P(e) {
  if (e !== void 0)
    return e === null ? null : {
      non_fungible_id: e.non_fungible_id
    };
}
function Ks(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function oe(e) {
  return $t(e);
}
function $t(e, t) {
  return e == null ? e : {
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(q)
  };
}
function se(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(P)
    };
}
function Us(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Bs(e) {
  return zt(e);
}
function zt(e, t) {
  return e == null ? e : {
    items: e.items.map(q)
  };
}
function $s(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(P)
    };
}
const zs = {
  Global: "Global"
};
function Ws(e) {
  let t = !0;
  return t = t && "aggregation_level" in e, t = t && "resource_address" in e, t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function Hs(e) {
  return ae(e);
}
function ae(e, t) {
  return e == null ? e : {
    aggregation_level: e.aggregation_level,
    resource_address: e.resource_address,
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function Wt(e) {
  if (e !== void 0)
    return e === null ? null : {
      aggregation_level: e.aggregation_level,
      resource_address: e.resource_address,
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function Qs(e) {
  let t = !0;
  return t = t && "vault_address" in e, t = t && "total_count" in e, t = t && "last_updated_at_state_version" in e, t;
}
function C(e) {
  return Ht(e);
}
function Ht(e, t) {
  return e == null ? e : {
    vault_address: e.vault_address,
    total_count: e.total_count,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function x(e) {
  if (e !== void 0)
    return e === null ? null : {
      vault_address: e.vault_address,
      total_count: e.total_count,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function Xs(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function ue(e) {
  return Qt(e);
}
function Qt(e, t) {
  return e == null ? e : {
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(C)
  };
}
function le(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(x)
    };
}
const Ys = {
  Vault: "Vault"
};
function Zs(e) {
  let t = !0;
  return t = t && "aggregation_level" in e, t = t && "resource_address" in e, t = t && "vaults" in e, t;
}
function js(e) {
  return ce(e);
}
function ce(e, t) {
  return e == null ? e : {
    aggregation_level: e.aggregation_level,
    resource_address: e.resource_address,
    vaults: ue(e.vaults)
  };
}
function Xt(e) {
  if (e !== void 0)
    return e === null ? null : {
      aggregation_level: e.aggregation_level,
      resource_address: e.resource_address,
      vaults: le(e.vaults)
    };
}
function V(e) {
  return Yt(e);
}
function Yt(e, t) {
  if (e == null)
    return e;
  switch (e.aggregation_level) {
    case "Global":
      return { ...ae(e), aggregation_level: "Global" };
    case "Vault":
      return { ...ce(e), aggregation_level: "Vault" };
    default:
      throw new Error(`No variant of NonFungibleResourcesCollectionItem exists with 'aggregation_level=${e.aggregation_level}'`);
  }
}
function k(e) {
  if (e !== void 0) {
    if (e === null)
      return null;
    switch (e.aggregation_level) {
      case "Global":
        return Wt(e);
      case "Vault":
        return Xt(e);
      default:
        throw new Error(`No variant of NonFungibleResourcesCollectionItem exists with 'aggregation_level=${e.aggregation_level}'`);
    }
  }
}
function ea(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Zt(e) {
  return jt(e);
}
function jt(e, t) {
  return e == null ? e : {
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(V)
  };
}
function er(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(k)
    };
}
function ta(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function ra(e) {
  return tr(e);
}
function tr(e, t) {
  return e == null ? e : {
    items: e.items.map(V)
  };
}
function na(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(k)
    };
}
function ia(e) {
  let t = !0;
  return t = t && "aggregation_level" in e, t = t && "resource_address" in e, t;
}
function oa(e) {
  return rr(e);
}
function rr(e, t) {
  return e == null ? e : {
    aggregation_level: f(e.aggregation_level),
    resource_address: e.resource_address
  };
}
function sa(e) {
  if (e !== void 0)
    return e === null ? null : {
      aggregation_level: e.aggregation_level,
      resource_address: e.resource_address
    };
}
const aa = {
  Global: "Global"
};
function ua(e) {
  let t = !0;
  return t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function la(e) {
  return nr(e);
}
function nr(e, t) {
  return e == null ? e : {
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version,
    aggregation_level: r(e, "aggregation_level") ? e.aggregation_level : void 0
  };
}
function ca(e) {
  if (e !== void 0)
    return e === null ? null : {
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version,
      aggregation_level: e.aggregation_level
    };
}
const da = {
  Vault: "Vault"
};
function _a(e) {
  let t = !0;
  return t = t && "vaults" in e, t;
}
function ga(e) {
  return ir(e);
}
function ir(e, t) {
  return e == null ? e : {
    vaults: ue(e.vaults),
    aggregation_level: r(e, "aggregation_level") ? e.aggregation_level : void 0
  };
}
function fa(e) {
  if (e !== void 0)
    return e === null ? null : {
      vaults: le(e.vaults),
      aggregation_level: e.aggregation_level
    };
}
function pa(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function ma(e) {
  return or(e);
}
function or(e, t) {
  return e == null ? e : {
    items: e.items.map(C)
  };
}
function ya(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(x)
    };
}
const Sa = {
  NotSyncedUpError: "NotSyncedUpError"
};
function Oa(e) {
  let t = !0;
  return t = t && "request_type" in e, t = t && "current_sync_delay_seconds" in e, t = t && "max_allowed_sync_delay_seconds" in e, t;
}
function Na(e) {
  return sr(e);
}
function sr(e, t) {
  return e == null ? e : {
    request_type: e.request_type,
    current_sync_delay_seconds: e.current_sync_delay_seconds,
    max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds,
    type: r(e, "type") ? e.type : void 0
  };
}
function Fa(e) {
  if (e !== void 0)
    return e === null ? null : {
      request_type: e.request_type,
      current_sync_delay_seconds: e.current_sync_delay_seconds,
      max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds,
      type: e.type
    };
}
const Ra = {
  EcdsaSecp256k1: "EcdsaSecp256k1"
};
function ba(e) {
  let t = !0;
  return t = t && "key_type" in e, t = t && "key_hex" in e, t;
}
function ha(e) {
  return de(e);
}
function de(e, t) {
  return e == null ? e : {
    key_type: e.key_type,
    key_hex: e.key_hex
  };
}
function ar(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_type: e.key_type,
      key_hex: e.key_hex
    };
}
const Ea = {
  EddsaEd25519: "EddsaEd25519"
};
function Ta(e) {
  let t = !0;
  return t = t && "key_type" in e, t = t && "key_hex" in e, t;
}
function Ia(e) {
  return _e(e);
}
function _e(e, t) {
  return e == null ? e : {
    key_type: e.key_type,
    key_hex: e.key_hex
  };
}
function ur(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_type: e.key_type,
      key_hex: e.key_hex
    };
}
function b(e) {
  return lr(e);
}
function lr(e, t) {
  if (e == null)
    return e;
  switch (e.key_type) {
    case "EcdsaSecp256k1":
      return { ...de(e), key_type: "EcdsaSecp256k1" };
    case "EddsaEd25519":
      return { ..._e(e), key_type: "EddsaEd25519" };
    default:
      throw new Error(`No variant of PublicKey exists with 'key_type=${e.key_type}'`);
  }
}
function h(e) {
  if (e !== void 0) {
    if (e === null)
      return null;
    switch (e.key_type) {
      case "EcdsaSecp256k1":
        return ar(e);
      case "EddsaEd25519":
        return ur(e);
      default:
        throw new Error(`No variant of PublicKey exists with 'key_type=${e.key_type}'`);
    }
  }
}
const Ja = {
  EcdsaSecp256k1: "EcdsaSecp256k1",
  EddsaEd25519: "EddsaEd25519"
};
function cr(e) {
  return dr(e);
}
function dr(e, t) {
  return e;
}
function wa(e) {
  return e;
}
function va(e) {
  let t = !0;
  return t = t && "key_type" in e, t;
}
function Da(e) {
  return _r(e);
}
function _r(e, t) {
  return e == null ? e : {
    key_type: cr(e.key_type)
  };
}
function Aa(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_type: e.key_type
    };
}
const qa = {
  EcdsaSecp256k1: "EcdsaSecp256k1"
};
function Pa(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function Ca(e) {
  return gr(e);
}
function gr(e, t) {
  return e == null ? e : {
    key_hex: e.key_hex,
    key_type: r(e, "key_type") ? e.key_type : void 0
  };
}
function xa(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_hex: e.key_hex,
      key_type: e.key_type
    };
}
const Va = {
  EddsaEd25519: "EddsaEd25519"
};
function ka(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function La(e) {
  return fr(e);
}
function fr(e, t) {
  return e == null ? e : {
    key_hex: e.key_hex,
    key_type: r(e, "key_type") ? e.key_type : void 0
  };
}
function Ma(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_hex: e.key_hex,
      key_type: e.key_type
    };
}
function Ga(e) {
  return !0;
}
function Ka(e) {
  return pr(e);
}
function pr(e, t) {
  return e == null ? e : {
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0
  };
}
function Ua(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor
    };
}
function Ba(e) {
  let t = !0;
  return t = t && "raw_hex" in e, t = t && "raw_json" in e, t;
}
function mr(e) {
  return yr(e);
}
function yr(e, t) {
  return e == null ? e : {
    raw_hex: e.raw_hex,
    raw_json: e.raw_json
  };
}
function Sr(e) {
  if (e !== void 0)
    return e === null ? null : {
      raw_hex: e.raw_hex,
      raw_json: e.raw_json
    };
}
function $a(e) {
  let t = !0;
  return t = t && "addresses" in e, t;
}
function za(e) {
  return Or(e);
}
function Or(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    addresses: e.addresses,
    aggregation_level: r(e, "aggregation_level") ? f(e.aggregation_level) : void 0
  };
}
function Nr(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      addresses: e.addresses,
      aggregation_level: e.aggregation_level
    };
}
function Wa(e) {
  let t = !0;
  return t = t && "addresses" in e, t;
}
function Ha(e) {
  return Fr(e);
}
function Fr(e, t) {
  return e == null ? e : {
    addresses: e.addresses,
    aggregation_level: r(e, "aggregation_level") ? f(e.aggregation_level) : void 0
  };
}
function Qa(e) {
  if (e !== void 0)
    return e === null ? null : {
      addresses: e.addresses,
      aggregation_level: e.aggregation_level
    };
}
function Xa(e) {
  return !0;
}
function Rr(e) {
  return br(e);
}
function br(e, t) {
  return e == null ? e : {
    parent_address: r(e, "parent_address") ? e.parent_address : void 0,
    owner_address: r(e, "owner_address") ? e.owner_address : void 0,
    global_address: r(e, "global_address") ? e.global_address : void 0
  };
}
function hr(e) {
  if (e !== void 0)
    return e === null ? null : {
      parent_address: e.parent_address,
      owner_address: e.owner_address,
      global_address: e.global_address
    };
}
const Ya = {
  Component: "Component"
};
function Za(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "blueprint_name" in e, t = t && "access_rules_chain" in e, t;
}
function ja(e) {
  return ge(e);
}
function ge(e, t) {
  return e == null ? e : {
    type: e.type,
    package_address: r(e, "package_address") ? e.package_address : void 0,
    blueprint_name: e.blueprint_name,
    state: r(e, "state") ? e.state : void 0,
    access_rules_chain: e.access_rules_chain,
    royalty_aggregator: r(e, "royalty_aggregator") ? F(e.royalty_aggregator) : void 0
  };
}
function Er(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      package_address: e.package_address,
      blueprint_name: e.blueprint_name,
      state: e.state,
      access_rules_chain: e.access_rules_chain,
      royalty_aggregator: y(e.royalty_aggregator)
    };
}
const eu = {
  FungibleResource: "FungibleResource"
};
function tu(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "divisibility" in e, t;
}
function ru(e) {
  return fe(e);
}
function fe(e, t) {
  return e == null ? e : {
    type: e.type,
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    divisibility: e.divisibility
  };
}
function Tr(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      divisibility: e.divisibility
    };
}
const nu = {
  FungibleVault: "FungibleVault"
};
function iu(e) {
  let t = !0;
  return t = t && "type" in e, t;
}
function ou(e) {
  return pe(e);
}
function pe(e, t) {
  return e == null ? e : {
    type: e.type
  };
}
function Ir(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
const su = {
  NonFungibleResource: "NonFungibleResource"
};
function au(e) {
  let t = !0;
  return t = t && "type" in e, t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "non_fungible_id_type" in e, t;
}
function uu(e) {
  return me(e);
}
function me(e, t) {
  return e == null ? e : {
    type: e.type,
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    non_fungible_id_type: R(e.non_fungible_id_type)
  };
}
function Jr(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      non_fungible_id_type: e.non_fungible_id_type
    };
}
const lu = {
  NonFungibleVault: "NonFungibleVault"
};
function cu(e) {
  let t = !0;
  return t = t && "type" in e, t;
}
function du(e) {
  return ye(e);
}
function ye(e, t) {
  return e == null ? e : {
    type: e.type
  };
}
function wr(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
const _u = {
  Package: "Package"
};
function gu(e) {
  let t = !0;
  return t = t && "type" in e, t;
}
function fu(e) {
  return Se(e);
}
function Se(e, t) {
  return e == null ? e : {
    type: e.type,
    code_hex: r(e, "code_hex") ? e.code_hex : void 0,
    royalty_aggregator: r(e, "royalty_aggregator") ? F(e.royalty_aggregator) : void 0
  };
}
function vr(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type,
      code_hex: e.code_hex,
      royalty_aggregator: y(e.royalty_aggregator)
    };
}
function Dr(e) {
  return Ar(e);
}
function Ar(e, t) {
  if (e == null)
    return e;
  switch (e.type) {
    case "Component":
      return { ...ge(e), type: "Component" };
    case "FungibleResource":
      return { ...fe(e), type: "FungibleResource" };
    case "FungibleVault":
      return { ...pe(e), type: "FungibleVault" };
    case "NonFungibleResource":
      return { ...me(e), type: "NonFungibleResource" };
    case "NonFungibleVault":
      return { ...ye(e), type: "NonFungibleVault" };
    case "Package":
      return { ...Se(e), type: "Package" };
    default:
      throw new Error(`No variant of StateEntityDetailsResponseItemDetails exists with 'type=${e.type}'`);
  }
}
function qr(e) {
  if (e !== void 0) {
    if (e === null)
      return null;
    switch (e.type) {
      case "Component":
        return Er(e);
      case "FungibleResource":
        return Tr(e);
      case "FungibleVault":
        return Ir(e);
      case "NonFungibleResource":
        return Jr(e);
      case "NonFungibleVault":
        return wr(e);
      case "Package":
        return vr(e);
      default:
        throw new Error(`No variant of StateEntityDetailsResponseItemDetails exists with 'type=${e.type}'`);
    }
  }
}
function pu(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "metadata" in e, t;
}
function Oe(e) {
  return Pr(e);
}
function Pr(e, t) {
  return e == null ? e : {
    address: e.address,
    fungible_resources: r(e, "fungible_resources") ? yt(e.fungible_resources) : void 0,
    non_fungible_resources: r(e, "non_fungible_resources") ? Zt(e.non_fungible_resources) : void 0,
    ancestor_identities: r(e, "ancestor_identities") ? Rr(e.ancestor_identities) : void 0,
    metadata: K(e.metadata),
    details: r(e, "details") ? Dr(e.details) : void 0
  };
}
function Ne(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      fungible_resources: Ot(e.fungible_resources),
      non_fungible_resources: er(e.non_fungible_resources),
      ancestor_identities: hr(e.ancestor_identities),
      metadata: U(e.metadata),
      details: qr(e.details)
    };
}
function mu(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t;
}
function Cr(e) {
  return xr(e);
}
function xr(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    items: e.items.map(Oe)
  };
}
function yu(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      items: e.items.map(Ne)
    };
}
function Su(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Ou(e) {
  return Vr(e);
}
function Vr(e, t) {
  return e == null ? e : {
    items: e.items.map(Oe)
  };
}
function Nu(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(Ne)
    };
}
const Fu = {
  Component: "Component"
};
function Ru(e) {
  let t = !0;
  return t = t && "blueprint_name" in e, t = t && "access_rules_chain" in e, t;
}
function bu(e) {
  return kr(e);
}
function kr(e, t) {
  return e == null ? e : {
    package_address: r(e, "package_address") ? e.package_address : void 0,
    blueprint_name: e.blueprint_name,
    state: r(e, "state") ? e.state : void 0,
    access_rules_chain: e.access_rules_chain,
    royalty_aggregator: r(e, "royalty_aggregator") ? F(e.royalty_aggregator) : void 0,
    type: r(e, "type") ? e.type : void 0
  };
}
function hu(e) {
  if (e !== void 0)
    return e === null ? null : {
      package_address: e.package_address,
      blueprint_name: e.blueprint_name,
      state: e.state,
      access_rules_chain: e.access_rules_chain,
      royalty_aggregator: y(e.royalty_aggregator),
      type: e.type
    };
}
const Eu = {
  FungibleResource: "FungibleResource"
};
function Tu(e) {
  let t = !0;
  return t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "divisibility" in e, t;
}
function Iu(e) {
  return Lr(e);
}
function Lr(e, t) {
  return e == null ? e : {
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    divisibility: e.divisibility,
    type: r(e, "type") ? e.type : void 0
  };
}
function Ju(e) {
  if (e !== void 0)
    return e === null ? null : {
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      divisibility: e.divisibility,
      type: e.type
    };
}
const wu = {
  FungibleVault: "FungibleVault"
};
function vu(e) {
  return !0;
}
function Du(e) {
  return Mr(e);
}
function Mr(e, t) {
  return e == null ? e : {
    type: r(e, "type") ? e.type : void 0
  };
}
function Au(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
const qu = {
  FungibleResource: "FungibleResource",
  NonFungibleResource: "NonFungibleResource",
  FungibleVault: "FungibleVault",
  NonFungibleVault: "NonFungibleVault",
  Package: "Package",
  Component: "Component"
};
function Gr(e) {
  return Kr(e);
}
function Kr(e, t) {
  return e;
}
function Pu(e) {
  return e;
}
function Cu(e) {
  let t = !0;
  return t = t && "type" in e, t;
}
function xu(e) {
  return Ur(e);
}
function Ur(e, t) {
  return e == null ? e : {
    type: Gr(e.type)
  };
}
function Vu(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
const ku = {
  NonFungibleResource: "NonFungibleResource"
};
function Lu(e) {
  let t = !0;
  return t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "non_fungible_id_type" in e, t;
}
function Mu(e) {
  return Br(e);
}
function Br(e, t) {
  return e == null ? e : {
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    non_fungible_id_type: R(e.non_fungible_id_type),
    type: r(e, "type") ? e.type : void 0
  };
}
function Gu(e) {
  if (e !== void 0)
    return e === null ? null : {
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      non_fungible_id_type: e.non_fungible_id_type,
      type: e.type
    };
}
const Ku = {
  NonFungibleVault: "NonFungibleVault"
};
function Uu(e) {
  return !0;
}
function Bu(e) {
  return $r(e);
}
function $r(e, t) {
  return e == null ? e : {
    type: r(e, "type") ? e.type : void 0
  };
}
function $u(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
const zu = {
  Package: "Package"
};
function Wu(e) {
  return !0;
}
function Hu(e) {
  return zr(e);
}
function zr(e, t) {
  return e == null ? e : {
    code_hex: r(e, "code_hex") ? e.code_hex : void 0,
    royalty_aggregator: r(e, "royalty_aggregator") ? F(e.royalty_aggregator) : void 0,
    type: r(e, "type") ? e.type : void 0
  };
}
function Qu(e) {
  if (e !== void 0)
    return e === null ? null : {
      code_hex: e.code_hex,
      royalty_aggregator: y(e.royalty_aggregator),
      type: e.type
    };
}
function Xu(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "resource_address" in e, t;
}
function Yu(e) {
  return Wr(e);
}
function Wr(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    resource_address: e.resource_address
  };
}
function Hr(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      resource_address: e.resource_address
    };
}
function Zu(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "resource_address" in e, t;
}
function ju(e) {
  return Qr(e);
}
function Qr(e, t) {
  return e == null ? e : {
    address: e.address,
    resource_address: e.resource_address
  };
}
function el(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      resource_address: e.resource_address
    };
}
function tl(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t = t && "resource_address" in e, t;
}
function Xr(e) {
  return Yr(e);
}
function Yr(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(w),
    address: e.address,
    resource_address: e.resource_address
  };
}
function rl(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(v),
      address: e.address,
      resource_address: e.resource_address
    };
}
function nl(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function il(e) {
  return Zr(e);
}
function Zr(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    aggregation_level: r(e, "aggregation_level") ? f(e.aggregation_level) : void 0
  };
}
function jr(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      aggregation_level: e.aggregation_level
    };
}
function ol(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function sl(e) {
  return en(e);
}
function en(e, t) {
  return e == null ? e : {
    address: e.address,
    aggregation_level: r(e, "aggregation_level") ? f(e.aggregation_level) : void 0
  };
}
function al(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      aggregation_level: e.aggregation_level
    };
}
function ul(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t;
}
function tn(e) {
  return rn(e);
}
function rn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(D),
    address: e.address
  };
}
function ll(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(A),
      address: e.address
    };
}
function cl(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function dl(e) {
  return nn(e);
}
function nn(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address
  };
}
function on(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address
    };
}
function _l(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function gl(e) {
  return sn(e);
}
function sn(e, t) {
  return e == null ? e : {
    address: e.address
  };
}
function fl(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address
    };
}
function pl(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t;
}
function an(e) {
  return un(e);
}
function un(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(I),
    address: e.address
  };
}
function ml(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(J),
      address: e.address
    };
}
function yl(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "vault_address" in e, t = t && "resource_address" in e, t;
}
function Sl(e) {
  return ln(e);
}
function ln(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    vault_address: e.vault_address,
    resource_address: e.resource_address
  };
}
function cn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      vault_address: e.vault_address,
      resource_address: e.resource_address
    };
}
function Ol(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "vault_address" in e, t = t && "resource_address" in e, t;
}
function Nl(e) {
  return dn(e);
}
function dn(e, t) {
  return e == null ? e : {
    address: e.address,
    vault_address: e.vault_address,
    resource_address: e.resource_address
  };
}
function Fl(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      vault_address: e.vault_address,
      resource_address: e.resource_address
    };
}
function Rl(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t = t && "resource_address" in e, t;
}
function _n(e) {
  return gn(e);
}
function gn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(q),
    address: e.address,
    resource_address: e.resource_address
  };
}
function bl(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(P),
      address: e.address,
      resource_address: e.resource_address
    };
}
function hl(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "resource_address" in e, t;
}
function El(e) {
  return fn(e);
}
function fn(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    resource_address: e.resource_address
  };
}
function pn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      resource_address: e.resource_address
    };
}
function Tl(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t = t && "resource_address" in e, t;
}
function mn(e) {
  return yn(e);
}
function yn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(C),
    address: e.address,
    resource_address: e.resource_address
  };
}
function Il(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(x),
      address: e.address,
      resource_address: e.resource_address
    };
}
function Jl(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function wl(e) {
  return Sn(e);
}
function Sn(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    aggregation_level: r(e, "aggregation_level") ? f(e.aggregation_level) : void 0
  };
}
function On(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      aggregation_level: e.aggregation_level
    };
}
function vl(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t;
}
function Nn(e) {
  return Fn(e);
}
function Fn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(V),
    address: e.address
  };
}
function Dl(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(k),
      address: e.address
    };
}
function Al(e) {
  let t = !0;
  return t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function ql(e) {
  return Rn(e);
}
function Rn(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0,
    resource_address: e.resource_address,
    non_fungible_ids: e.non_fungible_ids
  };
}
function bn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      resource_address: e.resource_address,
      non_fungible_ids: e.non_fungible_ids
    };
}
function Pl(e) {
  let t = !0;
  return t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function Cl(e) {
  return hn(e);
}
function hn(e, t) {
  return e == null ? e : {
    resource_address: e.resource_address,
    non_fungible_ids: e.non_fungible_ids
  };
}
function xl(e) {
  if (e !== void 0)
    return e === null ? null : {
      resource_address: e.resource_address,
      non_fungible_ids: e.non_fungible_ids
    };
}
function Vl(e) {
  let t = !0;
  return t = t && "non_fungible_id" in e, t = t && "mutable_data" in e, t = t && "last_updated_at_state_version" in e, t;
}
function Fe(e) {
  return En(e);
}
function En(e, t) {
  return e == null ? e : {
    non_fungible_id: e.non_fungible_id,
    mutable_data: mr(e.mutable_data),
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function Re(e) {
  if (e !== void 0)
    return e === null ? null : {
      non_fungible_id: e.non_fungible_id,
      mutable_data: Sr(e.mutable_data),
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function kl(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "resource_address" in e, t = t && "non_fungible_id_type" in e, t = t && "non_fungible_ids" in e, t;
}
function Tn(e) {
  return In(e);
}
function In(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    resource_address: e.resource_address,
    non_fungible_id_type: R(e.non_fungible_id_type),
    non_fungible_ids: e.non_fungible_ids.map(Fe)
  };
}
function Ll(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      resource_address: e.resource_address,
      non_fungible_id_type: e.non_fungible_id_type,
      non_fungible_ids: e.non_fungible_ids.map(Re)
    };
}
function Ml(e) {
  let t = !0;
  return t = t && "resource_address" in e, t = t && "non_fungible_id_type" in e, t = t && "non_fungible_ids" in e, t;
}
function Gl(e) {
  return Jn(e);
}
function Jn(e, t) {
  return e == null ? e : {
    resource_address: e.resource_address,
    non_fungible_id_type: R(e.non_fungible_id_type),
    non_fungible_ids: e.non_fungible_ids.map(Fe)
  };
}
function Kl(e) {
  if (e !== void 0)
    return e === null ? null : {
      resource_address: e.resource_address,
      non_fungible_id_type: e.non_fungible_id_type,
      non_fungible_ids: e.non_fungible_ids.map(Re)
    };
}
function Ul(e) {
  let t = !0;
  return t = t && "resource_address" in e, t;
}
function Bl(e) {
  return wn(e);
}
function wn(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0,
    resource_address: e.resource_address
  };
}
function vn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      resource_address: e.resource_address
    };
}
function $l(e) {
  let t = !0;
  return t = t && "resource_address" in e, t;
}
function zl(e) {
  return Dn(e);
}
function Dn(e, t) {
  return e == null ? e : {
    resource_address: e.resource_address
  };
}
function Wl(e) {
  if (e !== void 0)
    return e === null ? null : {
      resource_address: e.resource_address
    };
}
function Hl(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function An(e) {
  return qn(e);
}
function qn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    resource_address: e.resource_address,
    non_fungible_ids: oe(e.non_fungible_ids)
  };
}
function Ql(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      resource_address: e.resource_address,
      non_fungible_ids: se(e.non_fungible_ids)
    };
}
function Xl(e) {
  let t = !0;
  return t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function Yl(e) {
  return Pn(e);
}
function Pn(e, t) {
  return e == null ? e : {
    resource_address: e.resource_address,
    non_fungible_ids: oe(e.non_fungible_ids)
  };
}
function Zl(e) {
  if (e !== void 0)
    return e === null ? null : {
      resource_address: e.resource_address,
      non_fungible_ids: se(e.non_fungible_ids)
    };
}
function jl(e) {
  return !0;
}
function ec(e) {
  return Cn(e);
}
function Cn(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0
  };
}
function xn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor
    };
}
function tc(e) {
  return !0;
}
function rc(e) {
  return Vn(e);
}
function Vn(e, t) {
  return e == null ? e : {
    cursor: r(e, "cursor") ? e.cursor : void 0
  };
}
function nc(e) {
  if (e !== void 0)
    return e === null ? null : {
      cursor: e.cursor
    };
}
function ic(e) {
  let t = !0;
  return t = t && "stake" in e, t = t && "stake_percentage" in e, t = t && "key" in e, t;
}
function kn(e) {
  return Ln(e);
}
function Ln(e, t) {
  return e == null ? e : {
    stake: e.stake,
    stake_percentage: e.stake_percentage,
    key: b(e.key)
  };
}
function Mn(e) {
  if (e !== void 0)
    return e === null ? null : {
      stake: e.stake,
      stake_percentage: e.stake_percentage,
      key: h(e.key)
    };
}
function oc(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "current_stake" in e, t = t && "metadata" in e, t;
}
function be(e) {
  return Gn(e);
}
function Gn(e, t) {
  return e == null ? e : {
    address: e.address,
    state: r(e, "state") ? e.state : void 0,
    current_stake: e.current_stake,
    active_in_epoch: r(e, "active_in_epoch") ? kn(e.active_in_epoch) : void 0,
    metadata: K(e.metadata)
  };
}
function he(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      state: e.state,
      current_stake: e.current_stake,
      active_in_epoch: Mn(e.active_in_epoch),
      metadata: U(e.metadata)
    };
}
function sc(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Ee(e) {
  return Kn(e);
}
function Kn(e, t) {
  return e == null ? e : {
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(be)
  };
}
function Te(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(he)
    };
}
function ac(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "validators" in e, t;
}
function Un(e) {
  return Bn(e);
}
function Bn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    validators: Ee(e.validators)
  };
}
function uc(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      validators: Te(e.validators)
    };
}
function lc(e) {
  let t = !0;
  return t = t && "validators" in e, t;
}
function cc(e) {
  return $n(e);
}
function $n(e, t) {
  return e == null ? e : {
    validators: Ee(e.validators)
  };
}
function dc(e) {
  if (e !== void 0)
    return e === null ? null : {
      validators: Te(e.validators)
    };
}
const _c = {
  User: "User",
  EpochChange: "EpochChange",
  All: "All"
}, gc = {
  Asc: "Asc",
  Desc: "Desc"
};
function fc(e) {
  return !0;
}
function pc(e) {
  return zn(e);
}
function zn(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: r(e, "cursor") ? e.cursor : void 0,
    limit_per_page: r(e, "limit_per_page") ? e.limit_per_page : void 0,
    from_ledger_state: r(e, "from_ledger_state") ? _(e.from_ledger_state) : void 0,
    kind_filter: r(e, "kind_filter") ? e.kind_filter : void 0,
    order: r(e, "order") ? e.order : void 0
  };
}
function Wn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      from_ledger_state: g(e.from_ledger_state),
      kind_filter: e.kind_filter,
      order: e.order
    };
}
const mc = {
  User: "User",
  EpochChange: "EpochChange",
  All: "All"
}, yc = {
  Asc: "Asc",
  Desc: "Desc"
};
function Sc(e) {
  return !0;
}
function Oc(e) {
  return Hn(e);
}
function Hn(e, t) {
  return e == null ? e : {
    from_ledger_state: r(e, "from_ledger_state") ? _(e.from_ledger_state) : void 0,
    kind_filter: r(e, "kind_filter") ? e.kind_filter : void 0,
    order: r(e, "order") ? e.order : void 0
  };
}
function Nc(e) {
  if (e !== void 0)
    return e === null ? null : {
      from_ledger_state: g(e.from_ledger_state),
      kind_filter: e.kind_filter,
      order: e.order
    };
}
function Fc(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t;
}
function Qn(e) {
  return Xn(e);
}
function Xn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: r(e, "total_count") ? e.total_count : void 0,
    previous_cursor: r(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: r(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(E)
  };
}
function Rc(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(T)
    };
}
function bc(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function hc(e) {
  return Yn(e);
}
function Yn(e, t) {
  return e == null ? e : {
    at_ledger_state: r(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    intent_hash_hex: e.intent_hash_hex
  };
}
function Zn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: g(e.at_ledger_state),
      intent_hash_hex: e.intent_hash_hex
    };
}
function Ec(e) {
  let t = !0;
  return t = t && "raw_hex" in e, t = t && "receipt" in e, t = t && "referenced_global_entities" in e, t;
}
function Ie(e) {
  return jn(e);
}
function jn(e, t) {
  return e == null ? e : {
    raw_hex: e.raw_hex,
    receipt: e.receipt,
    referenced_global_entities: e.referenced_global_entities,
    message_hex: r(e, "message_hex") ? e.message_hex : void 0
  };
}
function Je(e) {
  if (e !== void 0)
    return e === null ? null : {
      raw_hex: e.raw_hex,
      receipt: e.receipt,
      referenced_global_entities: e.referenced_global_entities,
      message_hex: e.message_hex
    };
}
function Tc(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "transaction" in e, t = t && "details" in e, t;
}
function ei(e) {
  return ti(e);
}
function ti(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    transaction: E(e.transaction),
    details: Ie(e.details)
  };
}
function Ic(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      transaction: T(e.transaction),
      details: Je(e.details)
    };
}
function Jc(e) {
  let t = !0;
  return t = t && "transaction" in e, t = t && "details" in e, t;
}
function wc(e) {
  return ri(e);
}
function ri(e, t) {
  return e == null ? e : {
    transaction: E(e.transaction),
    details: Ie(e.details)
  };
}
function vc(e) {
  if (e !== void 0)
    return e === null ? null : {
      transaction: T(e.transaction),
      details: Je(e.details)
    };
}
function Dc(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t;
}
function ni(e) {
  return ii(e);
}
function ii(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state)
  };
}
function Ac(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state)
    };
}
const qc = {
  TransactionNotFoundError: "TransactionNotFoundError"
};
function Pc(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function Cc(e) {
  return oi(e);
}
function oi(e, t) {
  return e == null ? e : {
    intent_hash_hex: e.intent_hash_hex,
    type: r(e, "type") ? e.type : void 0
  };
}
function xc(e) {
  if (e !== void 0)
    return e === null ? null : {
      intent_hash_hex: e.intent_hash_hex,
      type: e.type
    };
}
function Vc(e) {
  let t = !0;
  return t = t && "unlimited_loan" in e, t = t && "assume_all_signature_proofs" in e, t = t && "permit_duplicate_intent_hash" in e, t = t && "permit_invalid_header_epoch" in e, t;
}
function si(e) {
  return ai(e);
}
function ai(e, t) {
  return e == null ? e : {
    unlimited_loan: e.unlimited_loan,
    assume_all_signature_proofs: e.assume_all_signature_proofs,
    permit_duplicate_intent_hash: e.permit_duplicate_intent_hash,
    permit_invalid_header_epoch: e.permit_invalid_header_epoch
  };
}
function ui(e) {
  if (e !== void 0)
    return e === null ? null : {
      unlimited_loan: e.unlimited_loan,
      assume_all_signature_proofs: e.assume_all_signature_proofs,
      permit_duplicate_intent_hash: e.permit_duplicate_intent_hash,
      permit_invalid_header_epoch: e.permit_invalid_header_epoch
    };
}
function kc(e) {
  let t = !0;
  return t = t && "manifest" in e, t = t && "start_epoch_inclusive" in e, t = t && "end_epoch_exclusive" in e, t = t && "cost_unit_limit" in e, t = t && "tip_percentage" in e, t = t && "nonce" in e, t = t && "signer_public_keys" in e, t = t && "flags" in e, t;
}
function Lc(e) {
  return li(e);
}
function li(e, t) {
  return e == null ? e : {
    manifest: e.manifest,
    blobs_hex: r(e, "blobs_hex") ? e.blobs_hex : void 0,
    start_epoch_inclusive: e.start_epoch_inclusive,
    end_epoch_exclusive: e.end_epoch_exclusive,
    notary_public_key: r(e, "notary_public_key") ? b(e.notary_public_key) : void 0,
    notary_as_signatory: r(e, "notary_as_signatory") ? e.notary_as_signatory : void 0,
    cost_unit_limit: e.cost_unit_limit,
    tip_percentage: e.tip_percentage,
    nonce: e.nonce,
    signer_public_keys: e.signer_public_keys.map(b),
    flags: si(e.flags)
  };
}
function ci(e) {
  if (e !== void 0)
    return e === null ? null : {
      manifest: e.manifest,
      blobs_hex: e.blobs_hex,
      start_epoch_inclusive: e.start_epoch_inclusive,
      end_epoch_exclusive: e.end_epoch_exclusive,
      notary_public_key: h(e.notary_public_key),
      notary_as_signatory: e.notary_as_signatory,
      cost_unit_limit: e.cost_unit_limit,
      tip_percentage: e.tip_percentage,
      nonce: e.nonce,
      signer_public_keys: e.signer_public_keys.map(h),
      flags: ui(e.flags)
    };
}
function Mc(e) {
  let t = !0;
  return t = t && "level" in e, t = t && "message" in e, t;
}
function di(e) {
  return _i(e);
}
function _i(e, t) {
  return e == null ? e : {
    level: e.level,
    message: e.message
  };
}
function gi(e) {
  if (e !== void 0)
    return e === null ? null : {
      level: e.level,
      message: e.message
    };
}
function Gc(e) {
  let t = !0;
  return t = t && "encoded_receipt" in e, t = t && "receipt" in e, t = t && "resource_changes" in e, t = t && "logs" in e, t;
}
function fi(e) {
  return pi(e);
}
function pi(e, t) {
  return e == null ? e : {
    encoded_receipt: e.encoded_receipt,
    receipt: e.receipt,
    resource_changes: e.resource_changes,
    logs: e.logs.map(di)
  };
}
function Kc(e) {
  if (e !== void 0)
    return e === null ? null : {
      encoded_receipt: e.encoded_receipt,
      receipt: e.receipt,
      resource_changes: e.resource_changes,
      logs: e.logs.map(gi)
    };
}
function Uc(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function Bc(e) {
  return mi(e);
}
function mi(e, t) {
  return e == null ? e : {
    intent_hash_hex: e.intent_hash_hex
  };
}
function yi(e) {
  if (e !== void 0)
    return e === null ? null : {
      intent_hash_hex: e.intent_hash_hex
    };
}
function $c(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function zc(e) {
  return Si(e);
}
function Si(e, t) {
  return e == null ? e : {
    intent_hash_hex: e.intent_hash_hex
  };
}
function Wc(e) {
  if (e !== void 0)
    return e === null ? null : {
      intent_hash_hex: e.intent_hash_hex
    };
}
function Hc(e) {
  let t = !0;
  return t = t && "payload_hash_hex" in e, t = t && "status" in e, t;
}
function we(e) {
  return Oi(e);
}
function Oi(e, t) {
  return e == null ? e : {
    payload_hash_hex: e.payload_hash_hex,
    status: N(e.status),
    error_message: r(e, "error_message") ? e.error_message : void 0
  };
}
function ve(e) {
  if (e !== void 0)
    return e === null ? null : {
      payload_hash_hex: e.payload_hash_hex,
      status: e.status,
      error_message: e.error_message
    };
}
function Qc(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "status" in e, t = t && "known_payloads" in e, t;
}
function Ni(e) {
  return Fi(e);
}
function Fi(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    status: N(e.status),
    known_payloads: e.known_payloads.map(we),
    error_message: r(e, "error_message") ? e.error_message : void 0
  };
}
function Xc(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      status: e.status,
      known_payloads: e.known_payloads.map(ve),
      error_message: e.error_message
    };
}
function Yc(e) {
  let t = !0;
  return t = t && "status" in e, t = t && "known_payloads" in e, t;
}
function Zc(e) {
  return Ri(e);
}
function Ri(e, t) {
  return e == null ? e : {
    status: N(e.status),
    known_payloads: e.known_payloads.map(we),
    error_message: r(e, "error_message") ? e.error_message : void 0
  };
}
function jc(e) {
  if (e !== void 0)
    return e === null ? null : {
      status: e.status,
      known_payloads: e.known_payloads.map(ve),
      error_message: e.error_message
    };
}
function ed(e) {
  let t = !0;
  return t = t && "notarized_transaction_hex" in e, t;
}
function td(e) {
  return bi(e);
}
function bi(e, t) {
  return e == null ? e : {
    notarized_transaction_hex: e.notarized_transaction_hex
  };
}
function hi(e) {
  if (e !== void 0)
    return e === null ? null : {
      notarized_transaction_hex: e.notarized_transaction_hex
    };
}
function rd(e) {
  let t = !0;
  return t = t && "duplicate" in e, t;
}
function Ei(e) {
  return Ti(e);
}
function Ti(e, t) {
  return e == null ? e : {
    duplicate: e.duplicate
  };
}
function nd(e) {
  if (e !== void 0)
    return e === null ? null : {
      duplicate: e.duplicate
    };
}
function id(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function od(e) {
  return Ii(e);
}
function Ii(e, t) {
  return e == null ? e : {
    items: e.items.map(be)
  };
}
function sd(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(he)
    };
}
class ad extends O {
  async entityFungibleResourceVaultPageRaw(t, n) {
    if (t.stateEntityFungibleResourceVaultsPageRequest === null || t.stateEntityFungibleResourceVaultsPageRequest === void 0)
      throw new d("stateEntityFungibleResourceVaultsPageRequest", "Required parameter requestParameters.stateEntityFungibleResourceVaultsPageRequest was null or undefined when calling entityFungibleResourceVaultPage.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/entity/page/fungible-vaults/",
      method: "POST",
      headers: o,
      query: i,
      body: Hr(t.stateEntityFungibleResourceVaultsPageRequest)
    }, n);
    return new u(s, (a) => Xr(a));
  }
  async entityFungibleResourceVaultPage(t, n) {
    return await (await this.entityFungibleResourceVaultPageRaw(t, n)).value();
  }
  async entityFungiblesPageRaw(t, n) {
    if (t.stateEntityFungiblesPageRequest === null || t.stateEntityFungiblesPageRequest === void 0)
      throw new d("stateEntityFungiblesPageRequest", "Required parameter requestParameters.stateEntityFungiblesPageRequest was null or undefined when calling entityFungiblesPage.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/entity/page/fungibles/",
      method: "POST",
      headers: o,
      query: i,
      body: jr(t.stateEntityFungiblesPageRequest)
    }, n);
    return new u(s, (a) => tn(a));
  }
  async entityFungiblesPage(t, n) {
    return await (await this.entityFungiblesPageRaw(t, n)).value();
  }
  async entityMetadataPageRaw(t, n) {
    if (t.stateEntityMetadataPageRequest === null || t.stateEntityMetadataPageRequest === void 0)
      throw new d("stateEntityMetadataPageRequest", "Required parameter requestParameters.stateEntityMetadataPageRequest was null or undefined when calling entityMetadataPage.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/entity/page/metadata",
      method: "POST",
      headers: o,
      query: i,
      body: on(t.stateEntityMetadataPageRequest)
    }, n);
    return new u(s, (a) => an(a));
  }
  async entityMetadataPage(t, n) {
    return await (await this.entityMetadataPageRaw(t, n)).value();
  }
  async entityNonFungibleIdsPageRaw(t, n) {
    if (t.stateEntityNonFungibleIdsPageRequest === null || t.stateEntityNonFungibleIdsPageRequest === void 0)
      throw new d("stateEntityNonFungibleIdsPageRequest", "Required parameter requestParameters.stateEntityNonFungibleIdsPageRequest was null or undefined when calling entityNonFungibleIdsPage.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/entity/page/non-fungible-vault/ids",
      method: "POST",
      headers: o,
      query: i,
      body: cn(t.stateEntityNonFungibleIdsPageRequest)
    }, n);
    return new u(s, (a) => _n(a));
  }
  async entityNonFungibleIdsPage(t, n) {
    return await (await this.entityNonFungibleIdsPageRaw(t, n)).value();
  }
  async entityNonFungibleResourceVaultPageRaw(t, n) {
    if (t.stateEntityNonFungibleResourceVaultsPageRequest === null || t.stateEntityNonFungibleResourceVaultsPageRequest === void 0)
      throw new d("stateEntityNonFungibleResourceVaultsPageRequest", "Required parameter requestParameters.stateEntityNonFungibleResourceVaultsPageRequest was null or undefined when calling entityNonFungibleResourceVaultPage.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/entity/page/non-fungible-vaults/",
      method: "POST",
      headers: o,
      query: i,
      body: pn(t.stateEntityNonFungibleResourceVaultsPageRequest)
    }, n);
    return new u(s, (a) => mn(a));
  }
  async entityNonFungibleResourceVaultPage(t, n) {
    return await (await this.entityNonFungibleResourceVaultPageRaw(t, n)).value();
  }
  async entityNonFungiblesPageRaw(t, n) {
    if (t.stateEntityNonFungiblesPageRequest === null || t.stateEntityNonFungiblesPageRequest === void 0)
      throw new d("stateEntityNonFungiblesPageRequest", "Required parameter requestParameters.stateEntityNonFungiblesPageRequest was null or undefined when calling entityNonFungiblesPage.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/entity/page/non-fungibles/",
      method: "POST",
      headers: o,
      query: i,
      body: On(t.stateEntityNonFungiblesPageRequest)
    }, n);
    return new u(s, (a) => Nn(a));
  }
  async entityNonFungiblesPage(t, n) {
    return await (await this.entityNonFungiblesPageRaw(t, n)).value();
  }
  async nonFungibleDataRaw(t, n) {
    if (t.stateNonFungibleDataRequest === null || t.stateNonFungibleDataRequest === void 0)
      throw new d("stateNonFungibleDataRequest", "Required parameter requestParameters.stateNonFungibleDataRequest was null or undefined when calling nonFungibleData.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/non-fungible/data",
      method: "POST",
      headers: o,
      query: i,
      body: bn(t.stateNonFungibleDataRequest)
    }, n);
    return new u(s, (a) => Tn(a));
  }
  async nonFungibleData(t, n) {
    return await (await this.nonFungibleDataRaw(t, n)).value();
  }
  async nonFungibleIdsRaw(t, n) {
    if (t.stateNonFungibleIdsRequest === null || t.stateNonFungibleIdsRequest === void 0)
      throw new d("stateNonFungibleIdsRequest", "Required parameter requestParameters.stateNonFungibleIdsRequest was null or undefined when calling nonFungibleIds.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/non-fungible/ids",
      method: "POST",
      headers: o,
      query: i,
      body: vn(t.stateNonFungibleIdsRequest)
    }, n);
    return new u(s, (a) => An(a));
  }
  async nonFungibleIds(t, n) {
    return await (await this.nonFungibleIdsRaw(t, n)).value();
  }
  async stateEntityDetailsRaw(t, n) {
    if (t.stateEntityDetailsRequest === null || t.stateEntityDetailsRequest === void 0)
      throw new d("stateEntityDetailsRequest", "Required parameter requestParameters.stateEntityDetailsRequest was null or undefined when calling stateEntityDetails.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/entity/details",
      method: "POST",
      headers: o,
      query: i,
      body: Nr(t.stateEntityDetailsRequest)
    }, n);
    return new u(s, (a) => Cr(a));
  }
  async stateEntityDetails(t, n) {
    return await (await this.stateEntityDetailsRaw(t, n)).value();
  }
  async stateValidatorsListRaw(t, n) {
    if (t.stateValidatorsListRequest === null || t.stateValidatorsListRequest === void 0)
      throw new d("stateValidatorsListRequest", "Required parameter requestParameters.stateValidatorsListRequest was null or undefined when calling stateValidatorsList.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/state/validators/list",
      method: "POST",
      headers: o,
      query: i,
      body: xn(t.stateValidatorsListRequest)
    }, n);
    return new u(s, (a) => Un(a));
  }
  async stateValidatorsList(t, n) {
    return await (await this.stateValidatorsListRaw(t, n)).value();
  }
}
class ud extends O {
  async gatewayStatusRaw(t) {
    const n = {}, i = {}, o = await this.request({
      path: "/status/gateway-status",
      method: "POST",
      headers: i,
      query: n
    }, t);
    return new u(o, (s) => vt(s));
  }
  async gatewayStatus(t) {
    return await (await this.gatewayStatusRaw(t)).value();
  }
  async networkConfigurationRaw(t) {
    const n = {}, i = {}, o = await this.request({
      path: "/status/network-configuration",
      method: "POST",
      headers: i,
      query: n
    }, t);
    return new u(o, (s) => Gt(s));
  }
  async networkConfiguration(t) {
    return await (await this.networkConfigurationRaw(t)).value();
  }
}
class ld extends O {
  async streamTransactionsRaw(t, n) {
    if (t.streamTransactionsRequest === null || t.streamTransactionsRequest === void 0)
      throw new d("streamTransactionsRequest", "Required parameter requestParameters.streamTransactionsRequest was null or undefined when calling streamTransactions.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/stream/transactions",
      method: "POST",
      headers: o,
      query: i,
      body: Wn(t.streamTransactionsRequest)
    }, n);
    return new u(s, (a) => Qn(a));
  }
  async streamTransactions(t, n) {
    return await (await this.streamTransactionsRaw(t, n)).value();
  }
}
class cd extends O {
  async transactionCommittedDetailsRaw(t, n) {
    if (t.transactionCommittedDetailsRequest === null || t.transactionCommittedDetailsRequest === void 0)
      throw new d("transactionCommittedDetailsRequest", "Required parameter requestParameters.transactionCommittedDetailsRequest was null or undefined when calling transactionCommittedDetails.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/transaction/committed-details",
      method: "POST",
      headers: o,
      query: i,
      body: Zn(t.transactionCommittedDetailsRequest)
    }, n);
    return new u(s, (a) => ei(a));
  }
  async transactionCommittedDetails(t, n) {
    return await (await this.transactionCommittedDetailsRaw(t, n)).value();
  }
  async transactionConstructionRaw(t) {
    const n = {}, i = {}, o = await this.request({
      path: "/transaction/construction",
      method: "POST",
      headers: i,
      query: n
    }, t);
    return new u(o, (s) => ni(s));
  }
  async transactionConstruction(t) {
    return await (await this.transactionConstructionRaw(t)).value();
  }
  async transactionPreviewRaw(t, n) {
    if (t.transactionPreviewRequest === null || t.transactionPreviewRequest === void 0)
      throw new d("transactionPreviewRequest", "Required parameter requestParameters.transactionPreviewRequest was null or undefined when calling transactionPreview.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/transaction/preview",
      method: "POST",
      headers: o,
      query: i,
      body: ci(t.transactionPreviewRequest)
    }, n);
    return new u(s, (a) => fi(a));
  }
  async transactionPreview(t, n) {
    return await (await this.transactionPreviewRaw(t, n)).value();
  }
  async transactionStatusRaw(t, n) {
    if (t.transactionStatusRequest === null || t.transactionStatusRequest === void 0)
      throw new d("transactionStatusRequest", "Required parameter requestParameters.transactionStatusRequest was null or undefined when calling transactionStatus.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/transaction/status",
      method: "POST",
      headers: o,
      query: i,
      body: yi(t.transactionStatusRequest)
    }, n);
    return new u(s, (a) => Ni(a));
  }
  async transactionStatus(t, n) {
    return await (await this.transactionStatusRaw(t, n)).value();
  }
  async transactionSubmitRaw(t, n) {
    if (t.transactionSubmitRequest === null || t.transactionSubmitRequest === void 0)
      throw new d("transactionSubmitRequest", "Required parameter requestParameters.transactionSubmitRequest was null or undefined when calling transactionSubmit.");
    const i = {}, o = {};
    o["Content-Type"] = "application/json";
    const s = await this.request({
      path: "/transaction/submit",
      method: "POST",
      headers: o,
      query: i,
      body: hi(t.transactionSubmitRequest)
    }, n);
    return new u(s, (a) => Ei(a));
  }
  async transactionSubmit(t, n) {
    return await (await this.transactionSubmitRaw(t, n)).value();
  }
}
export {
  Vi as AtLedgerStateMixinFromJSON,
  Me as AtLedgerStateMixinFromJSONTyped,
  ki as AtLedgerStateMixinToJSON,
  Pe as BASE_PATH,
  O as BaseAPI,
  qi as BlobApiResponse,
  wi as COLLECTION_FORMATS,
  E as CommittedTransactionInfoFromJSON,
  $e as CommittedTransactionInfoFromJSONTyped,
  T as CommittedTransactionInfoToJSON,
  Ce as Configuration,
  Bi as CursorLimitMixinFromJSON,
  ze as CursorLimitMixinFromJSONTyped,
  $i as CursorLimitMixinToJSON,
  xe as DefaultConfig,
  Xi as EntityMetadataCollectionAllOfFromJSON,
  Ze as EntityMetadataCollectionAllOfFromJSONTyped,
  Yi as EntityMetadataCollectionAllOfToJSON,
  K as EntityMetadataCollectionFromJSON,
  Ye as EntityMetadataCollectionFromJSONTyped,
  U as EntityMetadataCollectionToJSON,
  I as EntityMetadataItemFromJSON,
  Xe as EntityMetadataItemFromJSONTyped,
  J as EntityMetadataItemToJSON,
  ji as EntityMetadataItemValueAllOfFromJSON,
  je as EntityMetadataItemValueAllOfFromJSONTyped,
  eo as EntityMetadataItemValueAllOfToJSON,
  We as EntityMetadataItemValueFromJSON,
  He as EntityMetadataItemValueFromJSONTyped,
  Qe as EntityMetadataItemValueToJSON,
  so as EntityNotFoundErrorAllOfFromJSON,
  tt as EntityNotFoundErrorAllOfFromJSONTyped,
  ao as EntityNotFoundErrorAllOfToJSON,
  io as EntityNotFoundErrorAllOfTypeEnum,
  no as EntityNotFoundErrorFromJSON,
  B as EntityNotFoundErrorFromJSONTyped,
  et as EntityNotFoundErrorToJSON,
  to as EntityNotFoundErrorTypeEnum,
  wo as ErrorResponseFromJSON,
  _t as ErrorResponseFromJSONTyped,
  vo as ErrorResponseToJSON,
  Mo as FungibleResourcesCollectionAllOfFromJSON,
  Nt as FungibleResourcesCollectionAllOfFromJSONTyped,
  Go as FungibleResourcesCollectionAllOfToJSON,
  yt as FungibleResourcesCollectionFromJSON,
  St as FungibleResourcesCollectionFromJSONTyped,
  $o as FungibleResourcesCollectionItemBaseFromJSON,
  Rt as FungibleResourcesCollectionItemBaseFromJSONTyped,
  zo as FungibleResourcesCollectionItemBaseToJSON,
  D as FungibleResourcesCollectionItemFromJSON,
  mt as FungibleResourcesCollectionItemFromJSONTyped,
  Do as FungibleResourcesCollectionItemGloballyAggregatedAggregationLevelEnum,
  Wo as FungibleResourcesCollectionItemGloballyAggregatedAllOfAggregationLevelEnum,
  Qo as FungibleResourcesCollectionItemGloballyAggregatedAllOfFromJSON,
  bt as FungibleResourcesCollectionItemGloballyAggregatedAllOfFromJSONTyped,
  Xo as FungibleResourcesCollectionItemGloballyAggregatedAllOfToJSON,
  F as FungibleResourcesCollectionItemGloballyAggregatedFromJSON,
  j as FungibleResourcesCollectionItemGloballyAggregatedFromJSONTyped,
  y as FungibleResourcesCollectionItemGloballyAggregatedToJSON,
  A as FungibleResourcesCollectionItemToJSON,
  Co as FungibleResourcesCollectionItemVaultAggregatedAggregationLevelEnum,
  Yo as FungibleResourcesCollectionItemVaultAggregatedAllOfAggregationLevelEnum,
  jo as FungibleResourcesCollectionItemVaultAggregatedAllOfFromJSON,
  ht as FungibleResourcesCollectionItemVaultAggregatedAllOfFromJSONTyped,
  es as FungibleResourcesCollectionItemVaultAggregatedAllOfToJSON,
  Vo as FungibleResourcesCollectionItemVaultAggregatedFromJSON,
  re as FungibleResourcesCollectionItemVaultAggregatedFromJSONTyped,
  pt as FungibleResourcesCollectionItemVaultAggregatedToJSON,
  rs as FungibleResourcesCollectionItemVaultAggregatedVaultAllOfFromJSON,
  Et as FungibleResourcesCollectionItemVaultAggregatedVaultAllOfFromJSONTyped,
  ns as FungibleResourcesCollectionItemVaultAggregatedVaultAllOfToJSON,
  ee as FungibleResourcesCollectionItemVaultAggregatedVaultFromJSON,
  ft as FungibleResourcesCollectionItemVaultAggregatedVaultFromJSONTyped,
  w as FungibleResourcesCollectionItemVaultAggregatedVaultItemFromJSON,
  gt as FungibleResourcesCollectionItemVaultAggregatedVaultItemFromJSONTyped,
  v as FungibleResourcesCollectionItemVaultAggregatedVaultItemToJSON,
  te as FungibleResourcesCollectionItemVaultAggregatedVaultToJSON,
  Ot as FungibleResourcesCollectionToJSON,
  os as GatewayErrorBaseFromJSON,
  Tt as GatewayErrorBaseFromJSONTyped,
  ss as GatewayErrorBaseToJSON,
  lt as GatewayErrorFromJSON,
  ct as GatewayErrorFromJSONTyped,
  dt as GatewayErrorToJSON,
  us as GatewayInfoResponseKnownTargetFromJSON,
  It as GatewayInfoResponseKnownTargetFromJSONTyped,
  ls as GatewayInfoResponseKnownTargetToJSON,
  ne as GatewayInfoResponseReleaseInfoFromJSON,
  Jt as GatewayInfoResponseReleaseInfoFromJSONTyped,
  ie as GatewayInfoResponseReleaseInfoToJSON,
  ps as GatewayStatusResponseAllOfFromJSON,
  At as GatewayStatusResponseAllOfFromJSONTyped,
  ms as GatewayStatusResponseAllOfToJSON,
  vt as GatewayStatusResponseFromJSON,
  Dt as GatewayStatusResponseFromJSONTyped,
  gs as GatewayStatusResponseToJSON,
  Os as InternalServerErrorAllOfFromJSON,
  qt as InternalServerErrorAllOfFromJSONTyped,
  Ns as InternalServerErrorAllOfToJSON,
  ys as InternalServerErrorAllOfTypeEnum,
  co as InternalServerErrorFromJSON,
  $ as InternalServerErrorFromJSONTyped,
  rt as InternalServerErrorToJSON,
  uo as InternalServerErrorTypeEnum,
  bs as InvalidEntityErrorAllOfFromJSON,
  Pt as InvalidEntityErrorAllOfFromJSONTyped,
  hs as InvalidEntityErrorAllOfToJSON,
  Fs as InvalidEntityErrorAllOfTypeEnum,
  fo as InvalidEntityErrorFromJSON,
  z as InvalidEntityErrorFromJSONTyped,
  nt as InvalidEntityErrorToJSON,
  _o as InvalidEntityErrorTypeEnum,
  Is as InvalidRequestErrorAllOfFromJSON,
  Ct as InvalidRequestErrorAllOfFromJSONTyped,
  Js as InvalidRequestErrorAllOfToJSON,
  Es as InvalidRequestErrorAllOfTypeEnum,
  So as InvalidRequestErrorFromJSON,
  Q as InvalidRequestErrorFromJSONTyped,
  ot as InvalidRequestErrorToJSON,
  mo as InvalidRequestErrorTypeEnum,
  Ds as InvalidTransactionErrorAllOfFromJSON,
  xt as InvalidTransactionErrorAllOfFromJSONTyped,
  As as InvalidTransactionErrorAllOfToJSON,
  ws as InvalidTransactionErrorAllOfTypeEnum,
  Fo as InvalidTransactionErrorFromJSON,
  X as InvalidTransactionErrorFromJSONTyped,
  st as InvalidTransactionErrorToJSON,
  Oo as InvalidTransactionErrorTypeEnum,
  u as JSONApiResponse,
  l as LedgerStateFromJSON,
  wt as LedgerStateFromJSONTyped,
  Ps as LedgerStateMixinFromJSON,
  Vt as LedgerStateMixinFromJSONTyped,
  Cs as LedgerStateMixinToJSON,
  _ as LedgerStateSelectorFromJSON,
  Le as LedgerStateSelectorFromJSONTyped,
  g as LedgerStateSelectorToJSON,
  c as LedgerStateToJSON,
  Gt as NetworkConfigurationResponseFromJSON,
  Kt as NetworkConfigurationResponseFromJSONTyped,
  ks as NetworkConfigurationResponseToJSON,
  kt as NetworkConfigurationResponseWellKnownAddressesFromJSON,
  Lt as NetworkConfigurationResponseWellKnownAddressesFromJSONTyped,
  Mt as NetworkConfigurationResponseWellKnownAddressesToJSON,
  Ls as NonFungibleIdType,
  R as NonFungibleIdTypeFromJSON,
  Ut as NonFungibleIdTypeFromJSONTyped,
  Ms as NonFungibleIdTypeToJSON,
  Bs as NonFungibleIdsCollectionAllOfFromJSON,
  zt as NonFungibleIdsCollectionAllOfFromJSONTyped,
  $s as NonFungibleIdsCollectionAllOfToJSON,
  oe as NonFungibleIdsCollectionFromJSON,
  $t as NonFungibleIdsCollectionFromJSONTyped,
  q as NonFungibleIdsCollectionItemFromJSON,
  Bt as NonFungibleIdsCollectionItemFromJSONTyped,
  P as NonFungibleIdsCollectionItemToJSON,
  se as NonFungibleIdsCollectionToJSON,
  ra as NonFungibleResourcesCollectionAllOfFromJSON,
  tr as NonFungibleResourcesCollectionAllOfFromJSONTyped,
  na as NonFungibleResourcesCollectionAllOfToJSON,
  Zt as NonFungibleResourcesCollectionFromJSON,
  jt as NonFungibleResourcesCollectionFromJSONTyped,
  oa as NonFungibleResourcesCollectionItemBaseFromJSON,
  rr as NonFungibleResourcesCollectionItemBaseFromJSONTyped,
  sa as NonFungibleResourcesCollectionItemBaseToJSON,
  V as NonFungibleResourcesCollectionItemFromJSON,
  Yt as NonFungibleResourcesCollectionItemFromJSONTyped,
  zs as NonFungibleResourcesCollectionItemGloballyAggregatedAggregationLevelEnum,
  aa as NonFungibleResourcesCollectionItemGloballyAggregatedAllOfAggregationLevelEnum,
  la as NonFungibleResourcesCollectionItemGloballyAggregatedAllOfFromJSON,
  nr as NonFungibleResourcesCollectionItemGloballyAggregatedAllOfFromJSONTyped,
  ca as NonFungibleResourcesCollectionItemGloballyAggregatedAllOfToJSON,
  Hs as NonFungibleResourcesCollectionItemGloballyAggregatedFromJSON,
  ae as NonFungibleResourcesCollectionItemGloballyAggregatedFromJSONTyped,
  Wt as NonFungibleResourcesCollectionItemGloballyAggregatedToJSON,
  k as NonFungibleResourcesCollectionItemToJSON,
  Ys as NonFungibleResourcesCollectionItemVaultAggregatedAggregationLevelEnum,
  da as NonFungibleResourcesCollectionItemVaultAggregatedAllOfAggregationLevelEnum,
  ga as NonFungibleResourcesCollectionItemVaultAggregatedAllOfFromJSON,
  ir as NonFungibleResourcesCollectionItemVaultAggregatedAllOfFromJSONTyped,
  fa as NonFungibleResourcesCollectionItemVaultAggregatedAllOfToJSON,
  js as NonFungibleResourcesCollectionItemVaultAggregatedFromJSON,
  ce as NonFungibleResourcesCollectionItemVaultAggregatedFromJSONTyped,
  Xt as NonFungibleResourcesCollectionItemVaultAggregatedToJSON,
  ma as NonFungibleResourcesCollectionItemVaultAggregatedVaultAllOfFromJSON,
  or as NonFungibleResourcesCollectionItemVaultAggregatedVaultAllOfFromJSONTyped,
  ya as NonFungibleResourcesCollectionItemVaultAggregatedVaultAllOfToJSON,
  ue as NonFungibleResourcesCollectionItemVaultAggregatedVaultFromJSON,
  Qt as NonFungibleResourcesCollectionItemVaultAggregatedVaultFromJSONTyped,
  C as NonFungibleResourcesCollectionItemVaultAggregatedVaultItemFromJSON,
  Ht as NonFungibleResourcesCollectionItemVaultAggregatedVaultItemFromJSONTyped,
  x as NonFungibleResourcesCollectionItemVaultAggregatedVaultItemToJSON,
  le as NonFungibleResourcesCollectionItemVaultAggregatedVaultToJSON,
  er as NonFungibleResourcesCollectionToJSON,
  Na as NotSyncedUpErrorAllOfFromJSON,
  sr as NotSyncedUpErrorAllOfFromJSONTyped,
  Fa as NotSyncedUpErrorAllOfToJSON,
  Sa as NotSyncedUpErrorAllOfTypeEnum,
  ho as NotSyncedUpErrorFromJSON,
  Y as NotSyncedUpErrorFromJSONTyped,
  at as NotSyncedUpErrorToJSON,
  Ro as NotSyncedUpErrorTypeEnum,
  Da as PublicKeyBaseFromJSON,
  _r as PublicKeyBaseFromJSONTyped,
  Aa as PublicKeyBaseToJSON,
  Ca as PublicKeyEcdsaSecp256k1AllOfFromJSON,
  gr as PublicKeyEcdsaSecp256k1AllOfFromJSONTyped,
  qa as PublicKeyEcdsaSecp256k1AllOfKeyTypeEnum,
  xa as PublicKeyEcdsaSecp256k1AllOfToJSON,
  ha as PublicKeyEcdsaSecp256k1FromJSON,
  de as PublicKeyEcdsaSecp256k1FromJSONTyped,
  Ra as PublicKeyEcdsaSecp256k1KeyTypeEnum,
  ar as PublicKeyEcdsaSecp256k1ToJSON,
  La as PublicKeyEddsaEd25519AllOfFromJSON,
  fr as PublicKeyEddsaEd25519AllOfFromJSONTyped,
  Va as PublicKeyEddsaEd25519AllOfKeyTypeEnum,
  Ma as PublicKeyEddsaEd25519AllOfToJSON,
  Ia as PublicKeyEddsaEd25519FromJSON,
  _e as PublicKeyEddsaEd25519FromJSONTyped,
  Ea as PublicKeyEddsaEd25519KeyTypeEnum,
  ur as PublicKeyEddsaEd25519ToJSON,
  b as PublicKeyFromJSON,
  lr as PublicKeyFromJSONTyped,
  h as PublicKeyToJSON,
  Ja as PublicKeyType,
  cr as PublicKeyTypeFromJSON,
  dr as PublicKeyTypeFromJSONTyped,
  wa as PublicKeyTypeToJSON,
  d as RequiredError,
  Ko as ResourceAggregationLevel,
  f as ResourceAggregationLevelFromJSON,
  Ft as ResourceAggregationLevelFromJSONTyped,
  Uo as ResourceAggregationLevelToJSON,
  S as ResponseError,
  Ka as ResultSetCursorMixinFromJSON,
  pr as ResultSetCursorMixinFromJSONTyped,
  Ua as ResultSetCursorMixinToJSON,
  mr as ScryptoSborValueFromJSON,
  yr as ScryptoSborValueFromJSONTyped,
  Sr as ScryptoSborValueToJSON,
  ad as StateApi,
  Ha as StateEntityDetailsRequestAllOfFromJSON,
  Fr as StateEntityDetailsRequestAllOfFromJSONTyped,
  Qa as StateEntityDetailsRequestAllOfToJSON,
  za as StateEntityDetailsRequestFromJSON,
  Or as StateEntityDetailsRequestFromJSONTyped,
  Nr as StateEntityDetailsRequestToJSON,
  Ou as StateEntityDetailsResponseAllOfFromJSON,
  Vr as StateEntityDetailsResponseAllOfFromJSONTyped,
  Nu as StateEntityDetailsResponseAllOfToJSON,
  bu as StateEntityDetailsResponseComponentDetailsAllOfFromJSON,
  kr as StateEntityDetailsResponseComponentDetailsAllOfFromJSONTyped,
  hu as StateEntityDetailsResponseComponentDetailsAllOfToJSON,
  Fu as StateEntityDetailsResponseComponentDetailsAllOfTypeEnum,
  ja as StateEntityDetailsResponseComponentDetailsFromJSON,
  ge as StateEntityDetailsResponseComponentDetailsFromJSONTyped,
  Er as StateEntityDetailsResponseComponentDetailsToJSON,
  Ya as StateEntityDetailsResponseComponentDetailsTypeEnum,
  Cr as StateEntityDetailsResponseFromJSON,
  xr as StateEntityDetailsResponseFromJSONTyped,
  Iu as StateEntityDetailsResponseFungibleResourceDetailsAllOfFromJSON,
  Lr as StateEntityDetailsResponseFungibleResourceDetailsAllOfFromJSONTyped,
  Ju as StateEntityDetailsResponseFungibleResourceDetailsAllOfToJSON,
  Eu as StateEntityDetailsResponseFungibleResourceDetailsAllOfTypeEnum,
  ru as StateEntityDetailsResponseFungibleResourceDetailsFromJSON,
  fe as StateEntityDetailsResponseFungibleResourceDetailsFromJSONTyped,
  Tr as StateEntityDetailsResponseFungibleResourceDetailsToJSON,
  eu as StateEntityDetailsResponseFungibleResourceDetailsTypeEnum,
  Du as StateEntityDetailsResponseFungibleVaultDetailsAllOfFromJSON,
  Mr as StateEntityDetailsResponseFungibleVaultDetailsAllOfFromJSONTyped,
  Au as StateEntityDetailsResponseFungibleVaultDetailsAllOfToJSON,
  wu as StateEntityDetailsResponseFungibleVaultDetailsAllOfTypeEnum,
  ou as StateEntityDetailsResponseFungibleVaultDetailsFromJSON,
  pe as StateEntityDetailsResponseFungibleVaultDetailsFromJSONTyped,
  Ir as StateEntityDetailsResponseFungibleVaultDetailsToJSON,
  nu as StateEntityDetailsResponseFungibleVaultDetailsTypeEnum,
  Rr as StateEntityDetailsResponseItemAncestorIdentitiesFromJSON,
  br as StateEntityDetailsResponseItemAncestorIdentitiesFromJSONTyped,
  hr as StateEntityDetailsResponseItemAncestorIdentitiesToJSON,
  xu as StateEntityDetailsResponseItemDetailsBaseFromJSON,
  Ur as StateEntityDetailsResponseItemDetailsBaseFromJSONTyped,
  Vu as StateEntityDetailsResponseItemDetailsBaseToJSON,
  Dr as StateEntityDetailsResponseItemDetailsFromJSON,
  Ar as StateEntityDetailsResponseItemDetailsFromJSONTyped,
  qr as StateEntityDetailsResponseItemDetailsToJSON,
  qu as StateEntityDetailsResponseItemDetailsType,
  Gr as StateEntityDetailsResponseItemDetailsTypeFromJSON,
  Kr as StateEntityDetailsResponseItemDetailsTypeFromJSONTyped,
  Pu as StateEntityDetailsResponseItemDetailsTypeToJSON,
  Oe as StateEntityDetailsResponseItemFromJSON,
  Pr as StateEntityDetailsResponseItemFromJSONTyped,
  Ne as StateEntityDetailsResponseItemToJSON,
  Mu as StateEntityDetailsResponseNonFungibleResourceDetailsAllOfFromJSON,
  Br as StateEntityDetailsResponseNonFungibleResourceDetailsAllOfFromJSONTyped,
  Gu as StateEntityDetailsResponseNonFungibleResourceDetailsAllOfToJSON,
  ku as StateEntityDetailsResponseNonFungibleResourceDetailsAllOfTypeEnum,
  uu as StateEntityDetailsResponseNonFungibleResourceDetailsFromJSON,
  me as StateEntityDetailsResponseNonFungibleResourceDetailsFromJSONTyped,
  Jr as StateEntityDetailsResponseNonFungibleResourceDetailsToJSON,
  su as StateEntityDetailsResponseNonFungibleResourceDetailsTypeEnum,
  Bu as StateEntityDetailsResponseNonFungibleVaultDetailsAllOfFromJSON,
  $r as StateEntityDetailsResponseNonFungibleVaultDetailsAllOfFromJSONTyped,
  $u as StateEntityDetailsResponseNonFungibleVaultDetailsAllOfToJSON,
  Ku as StateEntityDetailsResponseNonFungibleVaultDetailsAllOfTypeEnum,
  du as StateEntityDetailsResponseNonFungibleVaultDetailsFromJSON,
  ye as StateEntityDetailsResponseNonFungibleVaultDetailsFromJSONTyped,
  wr as StateEntityDetailsResponseNonFungibleVaultDetailsToJSON,
  lu as StateEntityDetailsResponseNonFungibleVaultDetailsTypeEnum,
  Hu as StateEntityDetailsResponsePackageDetailsAllOfFromJSON,
  zr as StateEntityDetailsResponsePackageDetailsAllOfFromJSONTyped,
  Qu as StateEntityDetailsResponsePackageDetailsAllOfToJSON,
  zu as StateEntityDetailsResponsePackageDetailsAllOfTypeEnum,
  fu as StateEntityDetailsResponsePackageDetailsFromJSON,
  Se as StateEntityDetailsResponsePackageDetailsFromJSONTyped,
  vr as StateEntityDetailsResponsePackageDetailsToJSON,
  _u as StateEntityDetailsResponsePackageDetailsTypeEnum,
  yu as StateEntityDetailsResponseToJSON,
  ju as StateEntityFungibleResourceVaultsPageRequestAllOfFromJSON,
  Qr as StateEntityFungibleResourceVaultsPageRequestAllOfFromJSONTyped,
  el as StateEntityFungibleResourceVaultsPageRequestAllOfToJSON,
  Yu as StateEntityFungibleResourceVaultsPageRequestFromJSON,
  Wr as StateEntityFungibleResourceVaultsPageRequestFromJSONTyped,
  Hr as StateEntityFungibleResourceVaultsPageRequestToJSON,
  Xr as StateEntityFungibleResourceVaultsPageResponseFromJSON,
  Yr as StateEntityFungibleResourceVaultsPageResponseFromJSONTyped,
  rl as StateEntityFungibleResourceVaultsPageResponseToJSON,
  sl as StateEntityFungiblesPageRequestAllOfFromJSON,
  en as StateEntityFungiblesPageRequestAllOfFromJSONTyped,
  al as StateEntityFungiblesPageRequestAllOfToJSON,
  il as StateEntityFungiblesPageRequestFromJSON,
  Zr as StateEntityFungiblesPageRequestFromJSONTyped,
  jr as StateEntityFungiblesPageRequestToJSON,
  tn as StateEntityFungiblesPageResponseFromJSON,
  rn as StateEntityFungiblesPageResponseFromJSONTyped,
  ll as StateEntityFungiblesPageResponseToJSON,
  gl as StateEntityMetadataPageRequestAllOfFromJSON,
  sn as StateEntityMetadataPageRequestAllOfFromJSONTyped,
  fl as StateEntityMetadataPageRequestAllOfToJSON,
  dl as StateEntityMetadataPageRequestFromJSON,
  nn as StateEntityMetadataPageRequestFromJSONTyped,
  on as StateEntityMetadataPageRequestToJSON,
  an as StateEntityMetadataPageResponseFromJSON,
  un as StateEntityMetadataPageResponseFromJSONTyped,
  ml as StateEntityMetadataPageResponseToJSON,
  Nl as StateEntityNonFungibleIdsPageRequestAllOfFromJSON,
  dn as StateEntityNonFungibleIdsPageRequestAllOfFromJSONTyped,
  Fl as StateEntityNonFungibleIdsPageRequestAllOfToJSON,
  Sl as StateEntityNonFungibleIdsPageRequestFromJSON,
  ln as StateEntityNonFungibleIdsPageRequestFromJSONTyped,
  cn as StateEntityNonFungibleIdsPageRequestToJSON,
  _n as StateEntityNonFungibleIdsPageResponseFromJSON,
  gn as StateEntityNonFungibleIdsPageResponseFromJSONTyped,
  bl as StateEntityNonFungibleIdsPageResponseToJSON,
  El as StateEntityNonFungibleResourceVaultsPageRequestFromJSON,
  fn as StateEntityNonFungibleResourceVaultsPageRequestFromJSONTyped,
  pn as StateEntityNonFungibleResourceVaultsPageRequestToJSON,
  mn as StateEntityNonFungibleResourceVaultsPageResponseFromJSON,
  yn as StateEntityNonFungibleResourceVaultsPageResponseFromJSONTyped,
  Il as StateEntityNonFungibleResourceVaultsPageResponseToJSON,
  wl as StateEntityNonFungiblesPageRequestFromJSON,
  Sn as StateEntityNonFungiblesPageRequestFromJSONTyped,
  On as StateEntityNonFungiblesPageRequestToJSON,
  Nn as StateEntityNonFungiblesPageResponseFromJSON,
  Fn as StateEntityNonFungiblesPageResponseFromJSONTyped,
  Dl as StateEntityNonFungiblesPageResponseToJSON,
  Cl as StateNonFungibleDataRequestAllOfFromJSON,
  hn as StateNonFungibleDataRequestAllOfFromJSONTyped,
  xl as StateNonFungibleDataRequestAllOfToJSON,
  ql as StateNonFungibleDataRequestFromJSON,
  Rn as StateNonFungibleDataRequestFromJSONTyped,
  bn as StateNonFungibleDataRequestToJSON,
  Gl as StateNonFungibleDataResponseAllOfFromJSON,
  Jn as StateNonFungibleDataResponseAllOfFromJSONTyped,
  Kl as StateNonFungibleDataResponseAllOfToJSON,
  Tn as StateNonFungibleDataResponseFromJSON,
  In as StateNonFungibleDataResponseFromJSONTyped,
  Ll as StateNonFungibleDataResponseToJSON,
  Fe as StateNonFungibleDetailsResponseItemFromJSON,
  En as StateNonFungibleDetailsResponseItemFromJSONTyped,
  Re as StateNonFungibleDetailsResponseItemToJSON,
  zl as StateNonFungibleIdsRequestAllOfFromJSON,
  Dn as StateNonFungibleIdsRequestAllOfFromJSONTyped,
  Wl as StateNonFungibleIdsRequestAllOfToJSON,
  Bl as StateNonFungibleIdsRequestFromJSON,
  wn as StateNonFungibleIdsRequestFromJSONTyped,
  vn as StateNonFungibleIdsRequestToJSON,
  Yl as StateNonFungibleIdsResponseAllOfFromJSON,
  Pn as StateNonFungibleIdsResponseAllOfFromJSONTyped,
  Zl as StateNonFungibleIdsResponseAllOfToJSON,
  An as StateNonFungibleIdsResponseFromJSON,
  qn as StateNonFungibleIdsResponseFromJSONTyped,
  Ql as StateNonFungibleIdsResponseToJSON,
  rc as StateValidatorsListRequestAllOfFromJSON,
  Vn as StateValidatorsListRequestAllOfFromJSONTyped,
  nc as StateValidatorsListRequestAllOfToJSON,
  ec as StateValidatorsListRequestFromJSON,
  Cn as StateValidatorsListRequestFromJSONTyped,
  xn as StateValidatorsListRequestToJSON,
  cc as StateValidatorsListResponseAllOfFromJSON,
  $n as StateValidatorsListResponseAllOfFromJSONTyped,
  dc as StateValidatorsListResponseAllOfToJSON,
  Un as StateValidatorsListResponseFromJSON,
  Bn as StateValidatorsListResponseFromJSONTyped,
  uc as StateValidatorsListResponseToJSON,
  ud as StatusApi,
  ld as StreamApi,
  Oc as StreamTransactionsRequestAllOfFromJSON,
  Hn as StreamTransactionsRequestAllOfFromJSONTyped,
  mc as StreamTransactionsRequestAllOfKindFilterEnum,
  yc as StreamTransactionsRequestAllOfOrderEnum,
  Nc as StreamTransactionsRequestAllOfToJSON,
  pc as StreamTransactionsRequestFromJSON,
  zn as StreamTransactionsRequestFromJSONTyped,
  _c as StreamTransactionsRequestKindFilterEnum,
  gc as StreamTransactionsRequestOrderEnum,
  Wn as StreamTransactionsRequestToJSON,
  Qn as StreamTransactionsResponseFromJSON,
  Xn as StreamTransactionsResponseFromJSONTyped,
  Rc as StreamTransactionsResponseToJSON,
  Pi as TextApiResponse,
  Ge as TokenAmountFromJSON,
  Ke as TokenAmountFromJSONTyped,
  Ue as TokenAmountToJSON,
  cd as TransactionApi,
  hc as TransactionCommittedDetailsRequestFromJSON,
  Yn as TransactionCommittedDetailsRequestFromJSONTyped,
  Zn as TransactionCommittedDetailsRequestToJSON,
  wc as TransactionCommittedDetailsResponseAllOfFromJSON,
  ri as TransactionCommittedDetailsResponseAllOfFromJSONTyped,
  vc as TransactionCommittedDetailsResponseAllOfToJSON,
  Ie as TransactionCommittedDetailsResponseDetailsFromJSON,
  jn as TransactionCommittedDetailsResponseDetailsFromJSONTyped,
  Je as TransactionCommittedDetailsResponseDetailsToJSON,
  ei as TransactionCommittedDetailsResponseFromJSON,
  ti as TransactionCommittedDetailsResponseFromJSONTyped,
  Ic as TransactionCommittedDetailsResponseToJSON,
  ni as TransactionConstructionResponseFromJSON,
  ii as TransactionConstructionResponseFromJSONTyped,
  Ac as TransactionConstructionResponseToJSON,
  Cc as TransactionNotFoundErrorAllOfFromJSON,
  oi as TransactionNotFoundErrorAllOfFromJSONTyped,
  xc as TransactionNotFoundErrorAllOfToJSON,
  qc as TransactionNotFoundErrorAllOfTypeEnum,
  Io as TransactionNotFoundErrorFromJSON,
  Z as TransactionNotFoundErrorFromJSONTyped,
  ut as TransactionNotFoundErrorToJSON,
  Eo as TransactionNotFoundErrorTypeEnum,
  si as TransactionPreviewRequestFlagsFromJSON,
  ai as TransactionPreviewRequestFlagsFromJSONTyped,
  ui as TransactionPreviewRequestFlagsToJSON,
  Lc as TransactionPreviewRequestFromJSON,
  li as TransactionPreviewRequestFromJSONTyped,
  ci as TransactionPreviewRequestToJSON,
  fi as TransactionPreviewResponseFromJSON,
  pi as TransactionPreviewResponseFromJSONTyped,
  di as TransactionPreviewResponseLogsInnerFromJSON,
  _i as TransactionPreviewResponseLogsInnerFromJSONTyped,
  gi as TransactionPreviewResponseLogsInnerToJSON,
  Kc as TransactionPreviewResponseToJSON,
  Mi as TransactionStatus,
  N as TransactionStatusFromJSON,
  Be as TransactionStatusFromJSONTyped,
  zc as TransactionStatusRequestAllOfFromJSON,
  Si as TransactionStatusRequestAllOfFromJSONTyped,
  Wc as TransactionStatusRequestAllOfToJSON,
  Bc as TransactionStatusRequestFromJSON,
  mi as TransactionStatusRequestFromJSONTyped,
  yi as TransactionStatusRequestToJSON,
  Zc as TransactionStatusResponseAllOfFromJSON,
  Ri as TransactionStatusResponseAllOfFromJSONTyped,
  jc as TransactionStatusResponseAllOfToJSON,
  Ni as TransactionStatusResponseFromJSON,
  Fi as TransactionStatusResponseFromJSONTyped,
  we as TransactionStatusResponseKnownPayloadItemFromJSON,
  Oi as TransactionStatusResponseKnownPayloadItemFromJSONTyped,
  ve as TransactionStatusResponseKnownPayloadItemToJSON,
  Xc as TransactionStatusResponseToJSON,
  Gi as TransactionStatusToJSON,
  td as TransactionSubmitRequestFromJSON,
  bi as TransactionSubmitRequestFromJSONTyped,
  hi as TransactionSubmitRequestToJSON,
  Ei as TransactionSubmitResponseFromJSON,
  Ti as TransactionSubmitResponseFromJSONTyped,
  nd as TransactionSubmitResponseToJSON,
  W as ValidationErrorsAtPathFromJSON,
  it as ValidationErrorsAtPathFromJSONTyped,
  H as ValidationErrorsAtPathToJSON,
  od as ValidatorCollectionAllOfFromJSON,
  Ii as ValidatorCollectionAllOfFromJSONTyped,
  sd as ValidatorCollectionAllOfToJSON,
  Ee as ValidatorCollectionFromJSON,
  Kn as ValidatorCollectionFromJSONTyped,
  kn as ValidatorCollectionItemActiveInEpochFromJSON,
  Ln as ValidatorCollectionItemActiveInEpochFromJSONTyped,
  Mn as ValidatorCollectionItemActiveInEpochToJSON,
  be as ValidatorCollectionItemFromJSON,
  Gn as ValidatorCollectionItemFromJSONTyped,
  he as ValidatorCollectionItemToJSON,
  Te as ValidatorCollectionToJSON,
  Ai as VoidApiResponse,
  Di as canConsumeForm,
  r as exists,
  xi as instanceOfAtLedgerStateMixin,
  Ki as instanceOfCommittedTransactionInfo,
  Ui as instanceOfCursorLimitMixin,
  Hi as instanceOfEntityMetadataCollection,
  Qi as instanceOfEntityMetadataCollectionAllOf,
  Wi as instanceOfEntityMetadataItem,
  zi as instanceOfEntityMetadataItemValue,
  Zi as instanceOfEntityMetadataItemValueAllOf,
  ro as instanceOfEntityNotFoundError,
  oo as instanceOfEntityNotFoundErrorAllOf,
  Jo as instanceOfErrorResponse,
  ko as instanceOfFungibleResourcesCollection,
  Lo as instanceOfFungibleResourcesCollectionAllOf,
  Bo as instanceOfFungibleResourcesCollectionItemBase,
  Ao as instanceOfFungibleResourcesCollectionItemGloballyAggregated,
  Ho as instanceOfFungibleResourcesCollectionItemGloballyAggregatedAllOf,
  xo as instanceOfFungibleResourcesCollectionItemVaultAggregated,
  Zo as instanceOfFungibleResourcesCollectionItemVaultAggregatedAllOf,
  Po as instanceOfFungibleResourcesCollectionItemVaultAggregatedVault,
  ts as instanceOfFungibleResourcesCollectionItemVaultAggregatedVaultAllOf,
  qo as instanceOfFungibleResourcesCollectionItemVaultAggregatedVaultItem,
  is as instanceOfGatewayErrorBase,
  as as instanceOfGatewayInfoResponseKnownTarget,
  cs as instanceOfGatewayInfoResponseReleaseInfo,
  _s as instanceOfGatewayStatusResponse,
  fs as instanceOfGatewayStatusResponseAllOf,
  lo as instanceOfInternalServerError,
  Ss as instanceOfInternalServerErrorAllOf,
  go as instanceOfInvalidEntityError,
  Rs as instanceOfInvalidEntityErrorAllOf,
  yo as instanceOfInvalidRequestError,
  Ts as instanceOfInvalidRequestErrorAllOf,
  No as instanceOfInvalidTransactionError,
  vs as instanceOfInvalidTransactionErrorAllOf,
  ds as instanceOfLedgerState,
  qs as instanceOfLedgerStateMixin,
  Ci as instanceOfLedgerStateSelector,
  Vs as instanceOfNetworkConfigurationResponse,
  xs as instanceOfNetworkConfigurationResponseWellKnownAddresses,
  Ks as instanceOfNonFungibleIdsCollection,
  Us as instanceOfNonFungibleIdsCollectionAllOf,
  Gs as instanceOfNonFungibleIdsCollectionItem,
  ea as instanceOfNonFungibleResourcesCollection,
  ta as instanceOfNonFungibleResourcesCollectionAllOf,
  ia as instanceOfNonFungibleResourcesCollectionItemBase,
  Ws as instanceOfNonFungibleResourcesCollectionItemGloballyAggregated,
  ua as instanceOfNonFungibleResourcesCollectionItemGloballyAggregatedAllOf,
  Zs as instanceOfNonFungibleResourcesCollectionItemVaultAggregated,
  _a as instanceOfNonFungibleResourcesCollectionItemVaultAggregatedAllOf,
  Xs as instanceOfNonFungibleResourcesCollectionItemVaultAggregatedVault,
  pa as instanceOfNonFungibleResourcesCollectionItemVaultAggregatedVaultAllOf,
  Qs as instanceOfNonFungibleResourcesCollectionItemVaultAggregatedVaultItem,
  bo as instanceOfNotSyncedUpError,
  Oa as instanceOfNotSyncedUpErrorAllOf,
  va as instanceOfPublicKeyBase,
  ba as instanceOfPublicKeyEcdsaSecp256k1,
  Pa as instanceOfPublicKeyEcdsaSecp256k1AllOf,
  Ta as instanceOfPublicKeyEddsaEd25519,
  ka as instanceOfPublicKeyEddsaEd25519AllOf,
  Ga as instanceOfResultSetCursorMixin,
  Ba as instanceOfScryptoSborValue,
  $a as instanceOfStateEntityDetailsRequest,
  Wa as instanceOfStateEntityDetailsRequestAllOf,
  mu as instanceOfStateEntityDetailsResponse,
  Su as instanceOfStateEntityDetailsResponseAllOf,
  Za as instanceOfStateEntityDetailsResponseComponentDetails,
  Ru as instanceOfStateEntityDetailsResponseComponentDetailsAllOf,
  tu as instanceOfStateEntityDetailsResponseFungibleResourceDetails,
  Tu as instanceOfStateEntityDetailsResponseFungibleResourceDetailsAllOf,
  iu as instanceOfStateEntityDetailsResponseFungibleVaultDetails,
  vu as instanceOfStateEntityDetailsResponseFungibleVaultDetailsAllOf,
  pu as instanceOfStateEntityDetailsResponseItem,
  Xa as instanceOfStateEntityDetailsResponseItemAncestorIdentities,
  Cu as instanceOfStateEntityDetailsResponseItemDetailsBase,
  au as instanceOfStateEntityDetailsResponseNonFungibleResourceDetails,
  Lu as instanceOfStateEntityDetailsResponseNonFungibleResourceDetailsAllOf,
  cu as instanceOfStateEntityDetailsResponseNonFungibleVaultDetails,
  Uu as instanceOfStateEntityDetailsResponseNonFungibleVaultDetailsAllOf,
  gu as instanceOfStateEntityDetailsResponsePackageDetails,
  Wu as instanceOfStateEntityDetailsResponsePackageDetailsAllOf,
  Xu as instanceOfStateEntityFungibleResourceVaultsPageRequest,
  Zu as instanceOfStateEntityFungibleResourceVaultsPageRequestAllOf,
  tl as instanceOfStateEntityFungibleResourceVaultsPageResponse,
  nl as instanceOfStateEntityFungiblesPageRequest,
  ol as instanceOfStateEntityFungiblesPageRequestAllOf,
  ul as instanceOfStateEntityFungiblesPageResponse,
  cl as instanceOfStateEntityMetadataPageRequest,
  _l as instanceOfStateEntityMetadataPageRequestAllOf,
  pl as instanceOfStateEntityMetadataPageResponse,
  yl as instanceOfStateEntityNonFungibleIdsPageRequest,
  Ol as instanceOfStateEntityNonFungibleIdsPageRequestAllOf,
  Rl as instanceOfStateEntityNonFungibleIdsPageResponse,
  hl as instanceOfStateEntityNonFungibleResourceVaultsPageRequest,
  Tl as instanceOfStateEntityNonFungibleResourceVaultsPageResponse,
  Jl as instanceOfStateEntityNonFungiblesPageRequest,
  vl as instanceOfStateEntityNonFungiblesPageResponse,
  Al as instanceOfStateNonFungibleDataRequest,
  Pl as instanceOfStateNonFungibleDataRequestAllOf,
  kl as instanceOfStateNonFungibleDataResponse,
  Ml as instanceOfStateNonFungibleDataResponseAllOf,
  Vl as instanceOfStateNonFungibleDetailsResponseItem,
  Ul as instanceOfStateNonFungibleIdsRequest,
  $l as instanceOfStateNonFungibleIdsRequestAllOf,
  Hl as instanceOfStateNonFungibleIdsResponse,
  Xl as instanceOfStateNonFungibleIdsResponseAllOf,
  jl as instanceOfStateValidatorsListRequest,
  tc as instanceOfStateValidatorsListRequestAllOf,
  ac as instanceOfStateValidatorsListResponse,
  lc as instanceOfStateValidatorsListResponseAllOf,
  fc as instanceOfStreamTransactionsRequest,
  Sc as instanceOfStreamTransactionsRequestAllOf,
  Fc as instanceOfStreamTransactionsResponse,
  Li as instanceOfTokenAmount,
  bc as instanceOfTransactionCommittedDetailsRequest,
  Tc as instanceOfTransactionCommittedDetailsResponse,
  Jc as instanceOfTransactionCommittedDetailsResponseAllOf,
  Ec as instanceOfTransactionCommittedDetailsResponseDetails,
  Dc as instanceOfTransactionConstructionResponse,
  To as instanceOfTransactionNotFoundError,
  Pc as instanceOfTransactionNotFoundErrorAllOf,
  kc as instanceOfTransactionPreviewRequest,
  Vc as instanceOfTransactionPreviewRequestFlags,
  Gc as instanceOfTransactionPreviewResponse,
  Mc as instanceOfTransactionPreviewResponseLogsInner,
  Uc as instanceOfTransactionStatusRequest,
  $c as instanceOfTransactionStatusRequestAllOf,
  Qc as instanceOfTransactionStatusResponse,
  Yc as instanceOfTransactionStatusResponseAllOf,
  Hc as instanceOfTransactionStatusResponseKnownPayloadItem,
  ed as instanceOfTransactionSubmitRequest,
  rd as instanceOfTransactionSubmitResponse,
  po as instanceOfValidationErrorsAtPath,
  sc as instanceOfValidatorCollection,
  id as instanceOfValidatorCollectionAllOf,
  oc as instanceOfValidatorCollectionItem,
  ic as instanceOfValidatorCollectionItemActiveInEpoch,
  vi as mapValues,
  M as querystring
};
